{
  "cells": [
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "view-in-github",
        "colab_type": "text"
      },
      "source": [
        "<a href=\"https://colab.research.google.com/github/AdamLeeIT/DATA715/blob/main/pubmed_etl_exercise_2026.ipynb\" target=\"_parent\"><img src=\"https://colab.research.google.com/assets/colab-badge.svg\" alt=\"Open In Colab\"/></a>"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 1,
      "id": "086b0bd4-9473-4516-88e7-f5101c07d60f",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "086b0bd4-9473-4516-88e7-f5101c07d60f",
        "outputId": "559f36cf-b119-41c1-c2d6-2f0d0f1d6c0f"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Requirement already satisfied: Bio in /usr/local/lib/python3.12/dist-packages (1.8.1)\n",
            "Requirement already satisfied: biopython>=1.80 in /usr/local/lib/python3.12/dist-packages (from Bio) (1.86)\n",
            "Requirement already satisfied: gprofiler-official in /usr/local/lib/python3.12/dist-packages (from Bio) (1.0.0)\n",
            "Requirement already satisfied: mygene in /usr/local/lib/python3.12/dist-packages (from Bio) (3.2.2)\n",
            "Requirement already satisfied: pandas in /usr/local/lib/python3.12/dist-packages (from Bio) (2.2.2)\n",
            "Requirement already satisfied: pooch in /usr/local/lib/python3.12/dist-packages (from Bio) (1.9.0)\n",
            "Requirement already satisfied: requests in /usr/local/lib/python3.12/dist-packages (from Bio) (2.32.4)\n",
            "Requirement already satisfied: tqdm in /usr/local/lib/python3.12/dist-packages (from Bio) (4.67.3)\n",
            "Requirement already satisfied: numpy in /usr/local/lib/python3.12/dist-packages (from biopython>=1.80->Bio) (2.0.2)\n",
            "Requirement already satisfied: biothings-client>=0.2.6 in /usr/local/lib/python3.12/dist-packages (from mygene->Bio) (0.4.1)\n",
            "Requirement already satisfied: python-dateutil>=2.8.2 in /usr/local/lib/python3.12/dist-packages (from pandas->Bio) (2.9.0.post0)\n",
            "Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.12/dist-packages (from pandas->Bio) (2025.2)\n",
            "Requirement already satisfied: tzdata>=2022.7 in /usr/local/lib/python3.12/dist-packages (from pandas->Bio) (2025.3)\n",
            "Requirement already satisfied: platformdirs>=2.5.0 in /usr/local/lib/python3.12/dist-packages (from pooch->Bio) (4.9.2)\n",
            "Requirement already satisfied: packaging>=20.0 in /usr/local/lib/python3.12/dist-packages (from pooch->Bio) (26.0)\n",
            "Requirement already satisfied: charset_normalizer<4,>=2 in /usr/local/lib/python3.12/dist-packages (from requests->Bio) (3.4.4)\n",
            "Requirement already satisfied: idna<4,>=2.5 in /usr/local/lib/python3.12/dist-packages (from requests->Bio) (3.11)\n",
            "Requirement already satisfied: urllib3<3,>=1.21.1 in /usr/local/lib/python3.12/dist-packages (from requests->Bio) (2.5.0)\n",
            "Requirement already satisfied: certifi>=2017.4.17 in /usr/local/lib/python3.12/dist-packages (from requests->Bio) (2026.1.4)\n",
            "Requirement already satisfied: httpx>=0.22.0 in /usr/local/lib/python3.12/dist-packages (from biothings-client>=0.2.6->mygene->Bio) (0.28.1)\n",
            "Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.12/dist-packages (from python-dateutil>=2.8.2->pandas->Bio) (1.17.0)\n",
            "Requirement already satisfied: anyio in /usr/local/lib/python3.12/dist-packages (from httpx>=0.22.0->biothings-client>=0.2.6->mygene->Bio) (4.12.1)\n",
            "Requirement already satisfied: httpcore==1.* in /usr/local/lib/python3.12/dist-packages (from httpx>=0.22.0->biothings-client>=0.2.6->mygene->Bio) (1.0.9)\n",
            "Requirement already satisfied: h11>=0.16 in /usr/local/lib/python3.12/dist-packages (from httpcore==1.*->httpx>=0.22.0->biothings-client>=0.2.6->mygene->Bio) (0.16.0)\n",
            "Requirement already satisfied: typing_extensions>=4.5 in /usr/local/lib/python3.12/dist-packages (from anyio->httpx>=0.22.0->biothings-client>=0.2.6->mygene->Bio) (4.15.0)\n"
          ]
        }
      ],
      "source": [
        "pip install Bio"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 5,
      "id": "9a86a075-0c73-4afc-b0fb-794103800c2b",
      "metadata": {
        "id": "9a86a075-0c73-4afc-b0fb-794103800c2b"
      },
      "outputs": [],
      "source": [
        "from Bio import Entrez\n",
        "from Bio import Medline\n",
        "from tqdm import tqdm\n",
        "from itertools import combinations\n",
        "import json"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 7,
      "id": "5146c32a-f1e1-4378-beea-67cc52ae7b42",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "5146c32a-f1e1-4378-beea-67cc52ae7b42",
        "outputId": "a65d4a4b-d228-42e8-dc35-67804f3353f3"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Total number of publications that contain the term \"graph database\"\" AND 2025[pdat]: 25\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "100%|██████████| 1/1 [00:00<00:00,  1.59it/s]"
          ]
        },
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Complete.\n"
          ]
        },
        {
          "output_type": "stream",
          "name": "stderr",
          "text": [
            "\n"
          ]
        }
      ],
      "source": [
        "# Change this email to your email address\n",
        "Entrez.email = \"user@email.com\"\n",
        "\n",
        "keyword = '\"graph database\"\" AND 2025[pdat]'\n",
        "\n",
        "\n",
        "result = Entrez.read(Entrez.esearch(db=\"pubmed\", retmax=10, term=keyword))\n",
        "print(\n",
        "    \"Total number of publications that contain the term {}: {}\".format(\n",
        "        keyword, result[\"Count\"]\n",
        "    )\n",
        ")\n",
        "\n",
        "# Fetch all ids\n",
        "MAX_COUNT = result[\"Count\"]\n",
        "result = Entrez.read(\n",
        "    Entrez.esearch(db=\"pubmed\", retmax=result[\"Count\"], term=keyword)\n",
        "   #  Entrez.esearch(db=\"pubmed\", retmax=10, term=keyword)\n",
        "\n",
        ")\n",
        "\n",
        "ids = result[\"IdList\"]\n",
        "\n",
        "batch_size = 100\n",
        "batches = [ids[x: x + 100] for x in range(0, len(ids), batch_size)]\n",
        "\n",
        "record_list = []\n",
        "record_list2 = []\n",
        "\n",
        "for batch in tqdm(batches):\n",
        "    h = Entrez.efetch(db=\"pubmed\", id=batch, rettype=\"medline\", retmode=\"json\")\n",
        "    records = Medline.parse(h)\n",
        "    record_list.extend(list(records))\n",
        "print(\"Complete.\")"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 8,
      "id": "7e2b5cf0-5d4b-4b8a-b371-342315911b1f",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "7e2b5cf0-5d4b-4b8a-b371-342315911b1f",
        "outputId": "6aa25a46-67e9-4870-9747-4ad1bf1f6bc0"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "[{\"PMID\": \"41425057\", \"OWN\": \"NLM\", \"STAT\": \"PubMed-not-MEDLINE\", \"DCOM\": \"20251222\", \"LR\": \"20251224\", \"IS\": \"2624-8212 (Electronic) 2624-8212 (Linking)\", \"VI\": \"8\", \"DP\": \"2025\", \"TI\": \"NatureKG: an ontology and knowledge graph for nature finance with a Text2Cypher application.\", \"PG\": \"1693843\", \"LID\": \"10.3389/frai.2025.1693843 [doi] 1693843\", \"AB\": \"INTRODUCTION: Nature finance involves complex, multi-dimensional challenges that require analytical frameworks to assess risks, impacts, dependencies, and systemic resilience. Existing financial systems lack structured tools to map dependencies between natural capital and financial assets. To address this, we introduce NatureKG, the first ontology and instantiated knowledge graph (KG) specifically tailored to nature finance, aiming to support financial institutions in assessing environmental risks, impacts, and dependencies systematically. METHODS: We designed a domain ontology grounded in ENCORE, the Science-Based Targets Network (SBTN), and peer-reviewed literature. This ontology defines entities such as Actions, Drivers of Nature Loss, Value Chains, Evidence, and Sources. The ontology was instantiated into NatureKG within Neo4j, consisting of 320 nodes and 540 relationships curated by domain experts. As a proof of concept, we constructed a Text2Cypher dataset and fine-tuned three open-source large language models (Phi-3, LLaMA-3.1-8B, and Mistral-7B) to translate natural language queries into Cypher graph queries. The models were trained and evaluated under different dataset split strategies (paraphrase, cypher-level, and generalization) using metrics such as BLEU, exact match, execution accuracy, and Macro F1 scores. RESULTS: Phi-3 achieved the highest execution accuracy (0.21) and Macro F1 score (0.56), demonstrating better structural and reasoning capability under paraphrase and schema generalization splits. LLaMA-3.1-8B exhibited balanced performance, while Mistral-7B lagged across most metrics. The results indicate that smaller, fine-tuned models can generalize effectively in low-resource, domain-specific settings, validating the feasibility of LLM-assisted querying for nature finance. DISCUSSION: Despite modest initial accuracy, this feasibility study establishes a baseline for integrating domain-specific ontologies with AI systems. NatureKG offers a reusable foundation for representing environmental risks, dependencies, and interventions, with potential to enhance transparency and scalability in sustainable finance decision support. Future work should expand dataset diversity, sectoral coverage beyond the built environment, and refine model reasoning through larger, domain-aligned data catalogues.\", \"CI\": [\"Copyright (c) 2025 Kushwaha, Singh and Sheikh.\"], \"FAU\": [\"Kushwaha, Neetu\", \"Singh, Alok\", \"Sheikh, Hassan Aftab\"], \"AU\": [\"Kushwaha N\", \"Singh A\", \"Sheikh HA\"], \"AD\": [\"Smith School of Enterprise and the Environment, University of Oxford, Oxford, United Kingdom.\", \"NatureMind AI, Oxford, United Kingdom.\", \"Smith School of Enterprise and the Environment, University of Oxford, Oxford, United Kingdom.\", \"NatureMind AI, Oxford, United Kingdom.\", \"Data Science and AI in Creative Industry, Norwich University of the Arts, Norwich, United Kingdom.\", \"Smith School of Enterprise and the Environment, University of Oxford, Oxford, United Kingdom.\", \"NatureMind AI, Oxford, United Kingdom.\"], \"LA\": [\"eng\"], \"PT\": [\"Journal Article\"], \"DEP\": \"20251204\", \"PL\": \"Switzerland\", \"TA\": \"Front Artif Intell\", \"JT\": \"Frontiers in artificial intelligence\", \"JID\": \"101770551\", \"PMC\": \"PMC12713199\", \"OTO\": [\"NOTNLM\"], \"OT\": [\"Neo4j\", \"Text2Cypher\", \"cypher query\", \"graph database\", \"knowledge graphs\", \"large language models\", \"nature finance\", \"question answering\"], \"COIS\": [\"NK, AS, and HS were affiliated to NatureMind AI.\"], \"EDAT\": \"2025/12/22 18:09\", \"MHDA\": \"2025/12/22 18:10\", \"PMCR\": [\"2025/12/04\"], \"CRDT\": [\"2025/12/22 06:40\"], \"PHST\": [\"2025/08/27 00:00 [received]\", \"2025/10/31 00:00 [revised]\", \"2025/11/06 00:00 [accepted]\", \"2025/12/22 18:10 [medline]\", \"2025/12/22 18:09 [pubmed]\", \"2025/12/22 06:40 [entrez]\", \"2025/12/04 00:00 [pmc-release]\"], \"AID\": [\"10.3389/frai.2025.1693843 [doi]\"], \"PST\": \"epublish\", \"SO\": \"Front Artif Intell. 2025 Dec 4;8:1693843. doi: 10.3389/frai.2025.1693843. eCollection 2025.\"}, {\"PMID\": \"41369777\", \"OWN\": \"NLM\", \"STAT\": \"MEDLINE\", \"DCOM\": \"20251210\", \"LR\": \"20260203\", \"IS\": \"1432-2242 (Electronic) 0040-5752 (Linking)\", \"VI\": \"139\", \"IP\": \"1\", \"DP\": \"2025 Dec 10\", \"TI\": \"Comparative genomic analysis of QTL for resistance to Aphanomyces euteiches between pea, lentil, faba bean, and the model species Medicago truncatula.\", \"PG\": \"2\", \"LID\": \"10.1007/s00122-025-05095-4 [doi]\", \"AB\": \"QTL mapping and GWAS detected resistance QTL to Aphanomyces euteiches in faba bean, lentil, and Medicago truncatula. Weak genomic conservation between resistance QTL was identified between these legumes and pea. Aphanomyces root rot, caused by Aphanomyces euteiches, is a damaging disease affecting various legume species. Quantitative trait loci (QTL) for partial resistance have been mainly identified in pea, and to a lesser extent in lentil and Medicago truncatula. This study aimed to identify novel resistance loci from available lentil and faba bean populations, and examine genomic conservation of resistance QTL across legume host species. QTL mapping in the Pop2 faba bean recombinant inbred line (RIL) population and genome-wide association study (GWAS) in the AGILE lentil diversity panel were performed for resistance to A. euteiches under controlled conditions, using genotyping data previously reported. A previous QTL mapping in the LR3 M. truncatula RIL population was updated using 1,536 new SNPs (single-nucleotide polymorphisms). Synteny between resistance QTL to A. euteiches was analyzed based on gene orthology in QTL regions projected onto genomes, using the OrthoLegKB graph database. Four loci, including a major-effect QTL on chromosome 3, Ae-Vf3.1, were associated with resistance in faba bean. In lentil, six minor-effect GWAS-SNPs and two favorable haplotypes at Ae-Lc1.1 and Ae-Lc2.1 loci were identified. Updated analyses in M. truncatula narrowed to 8 Kb the interval of the major-effect locus AER1 and revealed three candidate genes. No synteny between major-effect QTL, detected in this study or previously reported in the literature, was identified across grain legume genomes. These results pave the way for translational genomics approaches facilitating resistance gene discovery and for resistance QTL deployment strategies in legume rotations to preserve their durability.\", \"CI\": [\"(c) 2025. The Author(s), under exclusive licence to Springer-Verlag GmbH Germany,\", \"part of Springer Nature.\"], \"FAU\": [\"Leprevost, Theo\", \"Imbert, Baptiste\", \"Lavaud, Clement\", \"Boutet, Gilles\", \"Miteul, Henri\", \"Leduc, Antoine\", \"Kreplak, Jonathan\", \"Phung, Ha-Trang\", \"Aubert, Gregoire\", \"Carrillo-Perdomo, Estefania\", \"Uhdre, Renan\", \"Sari, Hatice\", \"Bourland, Britton\", \"Caron, Carolyn T\", \"Tayeh, Nadim\", \"Ma, Yu\", \"Coyne, Clarice J\", \"Sugio, Akiko\", \"Pilet-Nayel, Marie-Laure\"], \"AU\": [\"Leprevost T\", \"Imbert B\", \"Lavaud C\", \"Boutet G\", \"Miteul H\", \"Leduc A\", \"Kreplak J\", \"Phung HT\", \"Aubert G\", \"Carrillo-Perdomo E\", \"Uhdre R\", \"Sari H\", \"Bourland B\", \"Caron CT\", \"Tayeh N\", \"Ma Y\", \"Coyne CJ\", \"Sugio A\", \"Pilet-Nayel ML\"], \"AUID\": [\"ORCID: 0000-0002-3424-4308\", \"ORCID: 0000-0002-8229-8693\", \"ORCID: 0009-0004-4372-0144\", \"ORCID: 0009-0006-6724-0641\", \"ORCID: 0000-0003-1509-2959\", \"ORCID: 0000-0001-5563-1932\", \"ORCID: 0009-0009-5352-2201\", \"ORCID: 0000-0002-1867-5730\", \"ORCID: 0000-0003-1772-2918\", \"ORCID: 0000-0003-2060-0241\", \"ORCID: 0000-0003-1230-5533\", \"ORCID: 0009-0009-8012-1796\", \"ORCID: 0000-0003-1251-7405\", \"ORCID: 0000-0003-0098-8322\", \"ORCID: 0000-0002-6840-8411\", \"ORCID: 0000-0003-1374-4618\", \"ORCID: 0000-0002-0970-144X\", \"ORCID: 0000-0002-4505-1907\"], \"AD\": [\"IGEPP, INRAE, Institut Agro, Univ Rennes, 35653, Le Rheu, France.\", \"Agroecologie, INRAE, Institut Agro, Univ. Bourgogne, Univ. Bourgogne Franche-Comte, 21000, Dijon, France.\", \"IGEPP, INRAE, Institut Agro, Univ Rennes, 35653, Le Rheu, France.\", \"IGEPP, INRAE, Institut Agro, Univ Rennes, 35653, Le Rheu, France.\", \"IGEPP, INRAE, Institut Agro, Univ Rennes, 35653, Le Rheu, France.\", \"IGEPP, INRAE, Institut Agro, Univ Rennes, 35653, Le Rheu, France.\", \"Agroecologie, INRAE, Institut Agro, Univ. Bourgogne, Univ. Bourgogne Franche-Comte, 21000, Dijon, France.\", \"Agroecologie, INRAE, Institut Agro, Univ. Bourgogne, Univ. Bourgogne Franche-Comte, 21000, Dijon, France.\", \"Agroecologie, INRAE, Institut Agro, Univ. Bourgogne, Univ. Bourgogne Franche-Comte, 21000, Dijon, France.\", \"Agroecologie, INRAE, Institut Agro, Univ. Bourgogne, Univ. Bourgogne Franche-Comte, 21000, Dijon, France.\", \"UMR AGAP Institut, INRAE, CIRAD, Universite de Montpellier, Institut Agro, 20230, San Giuliano, France.\", \"Department of Crop and Soil Sciences, Washington State University, Pullman, WA, 99164, USA.\", \"Department of Crop and Soil Sciences, Washington State University, Pullman, WA, 99164, USA.\", \"USDA-ARS Plant Germplasm Introduction & Testing Research, Washington State University, Pullman, WA, 99164, USA.\", \"Department of Plant Sciences, University of Saskatchewan, Saskatoon, S7N 5A8, Canada.\", \"Agroecologie, INRAE, Institut Agro, Univ. Bourgogne, Univ. Bourgogne Franche-Comte, 21000, Dijon, France.\", \"Department of Horticulture, Washington State University, Pullman, WA, 99164, USA.\", \"USDA-ARS Plant Germplasm Introduction & Testing Research, Washington State University, Pullman, WA, 99164, USA.\", \"IGEPP, INRAE, Institut Agro, Univ Rennes, 35653, Le Rheu, France.\", \"IGEPP, INRAE, Institut Agro, Univ Rennes, 35653, Le Rheu, France. marie-laure.pilet-nayel@inrae.fr.\"], \"LA\": [\"eng\"], \"GR\": [\"ANR-20-PCPA-0008/Agence Nationale de la Recherche/\", \"ANR-11-BTBR-0002/Agence Nationale de la Recherche/\", \"ANR-10-GENM-0007/Agence Nationale de la Recherche/\", \"101081878/HORIZON EUROPE Framework Programme/\"], \"PT\": [\"Comparative Study\", \"Journal Article\"], \"DEP\": \"20251210\", \"PL\": \"Germany\", \"TA\": \"Theor Appl Genet\", \"JT\": \"TAG. Theoretical and applied genetics. Theoretische und angewandte Genetik\", \"JID\": \"0145600\", \"SB\": \"IM\", \"MH\": [\"*Quantitative Trait Loci\", \"*Medicago truncatula/genetics/microbiology/parasitology\", \"*Disease Resistance/genetics\", \"*Aphanomyces/physiology\", \"*Lens Plant/genetics/microbiology\", \"*Plant Diseases/genetics/microbiology/parasitology\", \"Chromosome Mapping\", \"Polymorphism, Single Nucleotide\", \"*Pisum sativum/genetics/microbiology\", \"*Vicia faba/genetics/microbiology\", \"Genome-Wide Association Study\", \"Phenotype\", \"Genotype\"], \"COIS\": [\"Declarations. Conflict of interest: The authors declare that they have no\", \"conflict of interest. Ethical approval: Authors declare that the described\", \"experiments comply with the French laws.\"], \"EDAT\": \"2025/12/10 13:51\", \"MHDA\": \"2025/12/10 13:52\", \"CRDT\": [\"2025/12/10 11:14\"], \"PHST\": [\"2025/01/04 00:00 [received]\", \"2025/11/04 00:00 [accepted]\", \"2025/12/10 13:52 [medline]\", \"2025/12/10 13:51 [pubmed]\", \"2025/12/10 11:14 [entrez]\"], \"AID\": [\"10.1007/s00122-025-05095-4 [pii]\", \"10.1007/s00122-025-05095-4 [doi]\"], \"PST\": \"epublish\", \"SO\": \"Theor Appl Genet. 2025 Dec 10;139(1):2. doi: 10.1007/s00122-025-05095-4.\"}, {\"PMID\": \"41292650\", \"OWN\": \"NLM\", \"STAT\": \"PubMed-not-MEDLINE\", \"DCOM\": \"20251218\", \"LR\": \"20251219\", \"DP\": \"2025 Nov 15\", \"TI\": \"Finding Case Report Nuggets: A Web-Based Tool for Mining and Enhancing the Value of Clinical Case Reports.\", \"LID\": \"2025.11.13.25340162 [pii] 10.1101/2025.11.13.25340162 [doi]\", \"AB\": \"OBJECTIVES: Case reports are eyewitness reports of medical phenomena, such as adverse effects of treatments, outcomes of new surgical techniques, descriptions of rare diseases, unusual presentations of common diseases, or emerging infectious outbreaks. Although any single case report may be confounded, biased or erroneous, observations that are separately reported in multiple independent publications are more likely to be reliable, and so the accumulated evidence should have more value than any single report on its own. This notion led us to analyze the case reports literature in search of nuggets: collections of multiple case reports that describe similar main findings. MATERIALS AND METHODS: To identify nuggets in collections of case reports retrieved in PubMed queries, semantic similarities among the case reports were computed based on titles and main finding sentences extracted from the abstracts, and then grouped into communities with a graph database. The initial communities were then merged with a secondary hierarchical clustering process. RESULTS: Computed nuggets of size 4-100 articles are displayed along with large language model (LLM)-computed summaries, the title of the nugget's central article, and hyperlinks for viewing as well as export to our companion tool Anne O'Tate for further analysis. A variety of advanced options are also offered; users can optionally submit feedback on the quality of computed nuggets. DISCUSSION: Our free, public tool https://arrowsmith.psych.uic.edu/casereports facilitates the identification of nuggets and their summarization and mining. This should enhance the value of case report evidence and assist clinicians as well as those performing evidence syntheses of the published literature.\", \"FAU\": [\"Holt, Arthur W\", \"Smalheiser, Neil R\"], \"AU\": [\"Holt AW\", \"Smalheiser NR\"], \"AUID\": [\"ORCID: 0000-0002-6110-2997\", \"ORCID: 0000-0003-1079-3406\"], \"AD\": [\"Department of Psychiatry, University Of Illinois College of Medicine, 1601 W. Taylor Street, Chicago, IL 60612 USA.\", \"Department of Psychiatry, University Of Illinois College of Medicine, 1601 W. Taylor Street, Chicago, IL 60612 USA.\"], \"LA\": [\"eng\"], \"SI\": [\"Dryad/10.5061/dryad.5tb2rbpfb\"], \"GR\": [\"R01 LM014292/LM/NLM NIH HHS/United States\"], \"PT\": [\"Journal Article\", \"Preprint\"], \"DEP\": \"20251115\", \"PL\": \"United States\", \"TA\": \"medRxiv\", \"JT\": \"medRxiv : the preprint server for health sciences\", \"JID\": \"101767986\", \"PMC\": \"PMC12642711\", \"OTO\": [\"NOTNLM\"], \"OT\": [\"BERT embeddings\", \"Case reports\", \"PubMed\", \"data mining\", \"evidence based medicine\", \"graph database\", \"network analysis\", \"semantic similarity\"], \"COIS\": [\"CONFLICT OF INTEREST None declared.\"], \"EDAT\": \"2025/11/26 06:29\", \"MHDA\": \"2025/11/26 06:30\", \"PMCR\": [\"2025/11/24\"], \"CRDT\": [\"2025/11/26 04:35\"], \"PHST\": [\"2025/11/26 06:29 [pubmed]\", \"2025/11/26 06:30 [medline]\", \"2025/11/26 04:35 [entrez]\", \"2025/11/24 00:00 [pmc-release]\"], \"AID\": [\"2025.11.13.25340162 [pii]\", \"10.1101/2025.11.13.25340162 [doi]\"], \"PST\": \"epublish\", \"SO\": \"medRxiv [Preprint]. 2025 Nov 15:2025.11.13.25340162. doi: 10.1101/2025.11.13.25340162.\"}, {\"PMID\": \"41252800\", \"OWN\": \"NLM\", \"STAT\": \"MEDLINE\", \"DCOM\": \"20251211\", \"LR\": \"20251211\", \"IS\": \"1476-928X (Electronic) 1476-9271 (Linking)\", \"VI\": \"120\", \"IP\": \"Pt 1\", \"DP\": \"2026 Feb\", \"TI\": \"Using the Reactome Graph Database to generate pathway fingerprints for cancer-related genes, and using the fingerprints to gain insights into cancer, comorbidity, and cancer gene networks.\", \"PG\": \"108686\", \"LID\": \"S1476-9271(25)00347-0 [pii] 10.1016/j.compbiolchem.2025.108686 [doi]\", \"AB\": \"The Reactome Graph Database, which describes biological processes in 28 top-level pathways and facilitates efficient data- traversal and retrieval, has been used in a novel way to generate pathway fingerprints for 862 cancer driver- and 324 comorbidity-associated- genes; the latter set of genes cause Mendelian Diseases (MDs) which are comorbid with cancer. A pathway fingerprint presents pathway information about a gene in a novel way. It lists all pathways in which the gene participates, top-level-pathway-wise, the latter in decreasing order of importance. Data related to the above two sets of genes have also been mined from cBioPortal, OMIM, published literature on cancer-MD comorbidity and the NCI-compiled list of targeted therapies approved for cancers. Pathway fingerprints have connected- and added value to- data mined from the different repositories, and yielded insights into cancer and cancer-MD comorbidity. (1) Driver mutations in eight cancer datasets from cBioPortal, and allelic variants (AVs) from OMIM, were extracted for the cancer driver genes; the AVs were searched for among the driver mutations. Only by using the pathway fingerprints of driver genes was it possible to discern that the majority of cancer genes, which recruit AVs as somatic cancer mutations, play their most important role in the Signal Transduction top-level pathway. (2) OMIM-catalogued-, deleterious- AVs in comorbidity-associated genes also occur as somatic mutations in cancers, though infrequently. While in the study their occurrence appears not to be random, in the literature they are regarded as random passenger mutations. In over half the mutations, R is the mutated residue, R-->TER mutations being the most frequent. (3) A comparison of the pathway fingerprints of comorbidity-associated- and cancer- genes has shown that the largest number of shared pathways between the two sets of genes belongs to the Signal Transduction top-level pathway. Thus, mutated comorbidity-associated genes help cancer by adding to the burden of aberrant signal transduction. (4) Comorbidity-associated genes which cause Mendelian Inborn Errors of Metabolism tend not to share their disease-causing pathway with cancer genes while contributing to comorbidity. (5) Examining gene targets used in targeted cancer therapies, along with co-occurring genes of each target, yielded novel drug combinations for combination therapy; e.g., Desmopressin as an adjuvant for Dasatinib treatment. (6) Examining the pathway fingerprints of mutated driver genes in each cancer has shown that Signal Transduction and Gene expression are the most-affected top-level pathways in cancers, followed by Cell-Cycle, Developmental Biology and Immune System. (7) Targets for drug design have been identified from the set of genes causing the most prominent effects in the above top-level pathways. (8) Gene networks have been generated which show that: (i) the same source gene, in the same top-level pathway, interacts with similar target genes in different cancers; (ii) if the same source gene, in the same cancer, acts in different top-level pathways, one may identify top-level pathways that are connected by the source gene; (iii) if, in the same top-level pathway, in the same cancer, different source genes act, one may identify source genes that are connected.\", \"CI\": [\"Copyright (c) 2025 Elsevier Ltd. All rights reserved.\"], \"FAU\": [\"Iengar, Prathima\", \"Venkatraman, Prasanna\"], \"AU\": [\"Iengar P\", \"Venkatraman P\"], \"AD\": [\"Advanced Centre for Treatment, Research and Education in Cancer (ACTREC), Tata Memorial Centre, Sector 22, Kharghar, Navi Mumbai, Maharashtra 410210, India; Molecular Biophysics Unit, Division of Biological Sciences, Indian Institute of Science, Bangalore 560012, India. Electronic address: prathima@iisc.ac.in.\", \"Advanced Centre for Treatment, Research and Education in Cancer (ACTREC), Tata Memorial Centre, Sector 22, Kharghar, Navi Mumbai, Maharashtra 410210, India.\"], \"LA\": [\"eng\"], \"PT\": [\"Journal Article\"], \"DEP\": \"20250919\", \"PL\": \"England\", \"TA\": \"Comput Biol Chem\", \"JT\": \"Computational biology and chemistry\", \"JID\": \"101157394\", \"SB\": \"IM\", \"MH\": [\"*Neoplasms/genetics/epidemiology\", \"Humans\", \"*Gene Regulatory Networks\", \"Comorbidity\", \"*Databases, Genetic\", \"Mutation\"], \"OTO\": [\"NOTNLM\"], \"OT\": [\"Cancer gene networks\", \"Gene pathway fingerprint\", \"Identifying cancer drug targets\", \"Matrix of genes and reactions\", \"Pathways shared in cancer and comorbid Mendelian Diseases\", \"Targeted cancer therapy combinations\"], \"COIS\": [\"Declaration of Competing Interest None declared.\"], \"EDAT\": \"2025/11/18 18:30\", \"MHDA\": \"2025/12/11 07:44\", \"CRDT\": [\"2025/11/18 18:02\"], \"PHST\": [\"2025/03/06 00:00 [received]\", \"2025/09/01 00:00 [revised]\", \"2025/09/15 00:00 [accepted]\", \"2025/12/11 07:44 [medline]\", \"2025/11/18 18:30 [pubmed]\", \"2025/11/18 18:02 [entrez]\"], \"AID\": [\"S1476-9271(25)00347-0 [pii]\", \"10.1016/j.compbiolchem.2025.108686 [doi]\"], \"PST\": \"ppublish\", \"SO\": \"Comput Biol Chem. 2026 Feb;120(Pt 1):108686. doi: 10.1016/j.compbiolchem.2025.108686. Epub 2025 Sep 19.\"}, {\"PMID\": \"41173963\", \"OWN\": \"NLM\", \"STAT\": \"PubMed-not-MEDLINE\", \"LR\": \"20251104\", \"IS\": \"2045-2322 (Electronic) 2045-2322 (Linking)\", \"VI\": \"15\", \"IP\": \"1\", \"DP\": \"2025 Oct 31\", \"TI\": \"Research on gear shaving error analysis based on knowledge graph.\", \"PG\": \"38154\", \"LID\": \"10.1038/s41598-025-22007-0 [doi] 38154\", \"AB\": \"The influencing factors of shaving-induced mid-profile concavity error and tooth profile errors are numerous and complex, yet they lack effective interconnections, resulting in severe siloization issues. Existing literature research and practical manufacturing data have primarily accumulated the relationship between the influence of individual factors on gear shaving error and stored in the form of text dispersed. These discrete knowledges can't be systematically integrated and utilized and individual factor influence mechanism can't be effectively shared, resulting in the failure to quantitatively differentiate between the factors of the role of the mechanism, while the unclear underlying mechanisms obstruct the fundamental revelation of the mapping relationship between shaving-induced mid-profile concavity error and tooth profile errors. To address this, this study derives computational formulas for the improved gear shaving tooth profile cutting depth error and shaving allowance, solving the gear shaving tooth profile cutting depth error and shaving allowance by the shaving cutting parameter (radial feed), and analyzing the influence law of radial feed on the gear shaving tooth profile cutting depth error and shaving allowance. Experimental results demonstrate that when the degree of overlap ranges from 1.3 to 1.9, and the axis intersection angle error falls within 0.2 degrees ~0.5 degrees , selecting a radial feed of 44 ~ 50 mum simultaneously reduces tooth profile middle-concave error and tooth profile errors in workpiece gears. To resolve the lack of factor correlation, the Neo4j graph database was employed to construct a knowledge graph of influencing factors for shaving-induced mid-profile concavity error and tooth profile errors. This framework enables unified management of shaving error factors, the expression of correlations between knowledge entities, and relevant search and reasoning. It effectively resolves the issue of missing intrinsic connections among shaving error factors, realizes efficient retrieval and reasoning of knowledge regarding the intrinsic mechanism through which shaving error factors induce gear shaving errors, and thereby better facilitates the solution to tooth profile errors in shaving processes.\", \"CI\": [\"(c) 2025. The Author(s).\"], \"FAU\": [\"Men, Longlong\", \"Cai, Anjiang\"], \"AU\": [\"Men L\", \"Cai A\"], \"AD\": [\"School of Mechanical and Electrical Engineering, Xi'an University of Architecture and Technology, Xi'an, 710055, People's Republic of China.\", \"School of Mechanical and Electrical Engineering, Xi'an University of Architecture and Technology, Xi'an, 710055, People's Republic of China. men_l2@163.com.\"], \"LA\": [\"eng\"], \"GR\": [\"2023-YBGY-095/Key Research and Development Projects of Shaanxi Province,China/\", \"2023-YBGY-095/Key Research and Development Projects of Shaanxi Province,China/\"], \"PT\": [\"Journal Article\"], \"DEP\": \"20251031\", \"PL\": \"England\", \"TA\": \"Sci Rep\", \"JT\": \"Scientific reports\", \"JID\": \"101563288\", \"SB\": \"IM\", \"PMC\": \"PMC12579257\", \"OTO\": [\"NOTNLM\"], \"OT\": [\"Gear shaving tooth profile cutting depth error\", \"Knowledge graph\", \"Shaving-induced mid-profile concavity error\", \"Siloization issue\", \"Tooth profile error\"], \"COIS\": [\"Declarations. Competing interests: The authors declare no competing interests.\"], \"EDAT\": \"2025/11/01 05:37\", \"MHDA\": \"2025/11/01 05:38\", \"PMCR\": [\"2025/10/31\"], \"CRDT\": [\"2025/11/01 00:26\"], \"PHST\": [\"2025/07/11 00:00 [received]\", \"2025/09/25 00:00 [accepted]\", \"2025/11/01 05:38 [medline]\", \"2025/11/01 05:37 [pubmed]\", \"2025/11/01 00:26 [entrez]\", \"2025/10/31 00:00 [pmc-release]\"], \"AID\": [\"10.1038/s41598-025-22007-0 [pii]\", \"22007 [pii]\", \"10.1038/s41598-025-22007-0 [doi]\"], \"PST\": \"epublish\", \"SO\": \"Sci Rep. 2025 Oct 31;15(1):38154. doi: 10.1038/s41598-025-22007-0.\"}, {\"PMID\": \"41139924\", \"OWN\": \"NLM\", \"STAT\": \"MEDLINE\", \"DCOM\": \"20251027\", \"LR\": \"20251030\", \"IS\": \"1477-4054 (Electronic) 1467-5463 (Print) 1467-5463 (Linking)\", \"VI\": \"26\", \"IP\": \"5\", \"DP\": \"2025 Aug 31\", \"TI\": \"Expression graph network framework for biomarker discovery.\", \"LID\": \"10.1093/bib/bbaf559 [doi] bbaf559\", \"AB\": \"Biomarker discovery for complex diseases, such as cancer, hinges on uncovering molecular signatures that capture intricate, interconnected relationships within biological data-a challenge that traditional statistical and machine learning methods often fail to meet due to the complexity of high-dimensional gene expression profiles. To overcome this, we introduce the expression graph network framework (EGNF). This cutting-edge graph-based approach integrates graph neural networks with network-based feature engineering to enhance the predictive identification of biomarkers. EGNF constructs biologically informed networks by combining gene expression data and clinical attributes within a graph database, utilizing hierarchical clustering to generate dynamic, patient-specific representations of molecular interactions. Leveraging graph learning techniques, including graph convolutional networks and graph attention networks, our framework identifies statistically significant and biologically relevant gene modules for classification. Validated across three independent datasets consisting of contrasting tumor types and clinical scenarios, EGNF consistently outperforms traditional machine learning models, achieving superior classification accuracy and interpretability. Notably, it delivers perfect separation between normal and tumor samples while excelling in nuanced tasks such as classifying disease progression and predicting treatment outcomes. This scalable, interpretable, and robust framework provides a powerful tool for biomarker discovery, with wide-ranging applications in precision medicine and the elucidation of disease mechanisms across diverse clinical contexts.\", \"CI\": [\"(c) The Author(s) 2025. Published by Oxford University Press.\"], \"FAU\": [\"Liu, Yang\", \"Huse, Jason\", \"Kannan, Kasthuri\"], \"AU\": [\"Liu Y\", \"Huse J\", \"Kannan K\"], \"AD\": [\"Department of Translational Molecular Pathology, University of Texas MD Anderson Cancer Center, 2130 W Holcombe Blvd, Texas 77030, United States.\", \"Department of Biostatistics and Data Science, University of Texas Health Science Center at Houston, 1200 Pressler Street, Texas 77030, United States.\", \"Department of Translational Molecular Pathology, University of Texas MD Anderson Cancer Center, 2130 W Holcombe Blvd, Texas 77030, United States.\", \"Department of Pathology, University of Texas MD Anderson Cancer Center, 1515 Holcombe Blvd, Texas 77030, United States.\", \"Department of Translational Molecular Pathology, University of Texas MD Anderson Cancer Center, 2130 W Holcombe Blvd, Texas 77030, United States.\"], \"AUID\": [\"ORCID: 0000-0002-8479-0657\"], \"LA\": [\"eng\"], \"GR\": [\"MD Anderson Moonshot Program/\"], \"PT\": [\"Journal Article\"], \"PL\": \"England\", \"TA\": \"Brief Bioinform\", \"JT\": \"Briefings in bioinformatics\", \"JID\": \"100912837\", \"RN\": [\"0 (Biomarkers, Tumor)\", \"0 (Biomarkers)\"], \"SB\": \"IM\", \"MH\": [\"Humans\", \"*Biomarkers, Tumor/genetics\", \"Machine Learning\", \"*Neural Networks, Computer\", \"*Neoplasms/genetics/metabolism\", \"*Computational Biology/methods\", \"Gene Expression Profiling\", \"*Gene Regulatory Networks\", \"Biomarkers\"], \"PMC\": \"PMC12554635\", \"OTO\": [\"NOTNLM\"], \"OT\": [\"EGNF\", \"biomarker\", \"graph neural network\"], \"EDAT\": \"2025/10/27 13:28\", \"MHDA\": \"2025/10/27 13:29\", \"PMCR\": [\"2025/10/27\"], \"CRDT\": [\"2025/10/27 01:52\"], \"PHST\": [\"2025/07/09 00:00 [received]\", \"2025/08/18 00:00 [revised]\", \"2025/09/30 00:00 [accepted]\", \"2025/10/27 13:29 [medline]\", \"2025/10/27 13:28 [pubmed]\", \"2025/10/27 01:52 [entrez]\", \"2025/10/27 00:00 [pmc-release]\"], \"AID\": [\"8303424 [pii]\", \"bbaf559 [pii]\", \"10.1093/bib/bbaf559 [doi]\"], \"PST\": \"ppublish\", \"SO\": \"Brief Bioinform. 2025 Aug 31;26(5):bbaf559. doi: 10.1093/bib/bbaf559.\"}, {\"PMID\": \"41030660\", \"OWN\": \"NLM\", \"STAT\": \"PubMed-not-MEDLINE\", \"DCOM\": \"20251001\", \"LR\": \"20251003\", \"IS\": \"2352-3409 (Electronic) 2352-3409 (Linking)\", \"VI\": \"62\", \"DP\": \"2025 Oct\", \"TI\": \"A knowledge graph dataset for broiler farming automatically constructed based on a large language model.\", \"PG\": \"112018\", \"LID\": \"10.1016/j.dib.2025.112018 [doi] 112018\", \"AB\": \"With the rapid advancement of artificial intelligence, intelligent farming has become a key trend in modern agriculture. In particular, the application of intelligent systems in broiler farming is essential for enhancing production efficiency and optimizing management practices. Broiler farming is a complex process involving multiple interrelated components. However, existing knowledge graphs primarily focus on disease and prevention, making it difficult to capture the intricate interdependencies within the farming process. This limits the effectiveness of knowledge-based support in decision-making. To develop a high-quality broiler farming knowledge system, this study adopts large language modeling technology to integrate a Chinese corpus and construct a comprehensive knowledge graph dataset covering four core dimensions: broiler breeds, farming environment, feeding management, and disease prevention. The construction of the dataset involved three key stages. First, text scanning was used to extract information from farming-related literature, while web crawlers collected data from authoritative online sources. The data were then cleaned and manually validated to ensure accuracy and consistency. Second, the DeepKE knowledge extraction framework is used to automatically extract triples related to broiler farming from the text. These are then used as prompts to guide large-scale pre-trained language models (LLMs) to complete and optimize the knowledge, ultimately constructing a relatively complete knowledge graph of broiler farming. Finally, the structured knowledge was stored in a Neo4j graph database to support efficient querying and reasoning. The dataset not only provides researchers and farms with multidimensional knowledge of the broiler farming domain, but also supports visual management and analysis, enables data-driven inference through large models, and offers new approaches to optimize farming strategies and enhance production efficiency.\", \"CI\": [\"(c) 2025 The Authors.\"], \"FAU\": [\"Ma, Nan\", \"Kong, Fantao\", \"Liu, Jifang\", \"Zhang, Chenyang\", \"Zhao, Chenxv\", \"Cao, Shanshan\", \"Sun, Wei\"], \"AU\": [\"Ma N\", \"Kong F\", \"Liu J\", \"Zhang C\", \"Zhao C\", \"Cao S\", \"Sun W\"], \"AD\": [\"College of Computer and Information Engineering, Xinjiang Agricultural University, Urumqi 830052, China.\", \"Agricultural Information Institute of CAAS, Beijing 100081, China.\", \"Agricultural Economics and Development Institute of CAAS, Beijing 100081, China.\", \"National Nanfan Research Institute of CAAS, Sanya, 572024, China.\", \"Agricultural Information Institute of CAAS, Beijing 100081, China.\", \"National Nanfan Research Institute of CAAS, Sanya, 572024, China.\", \"College of Computer and Information Engineering, Xinjiang Agricultural University, Urumqi 830052, China.\", \"Agricultural Information Institute of CAAS, Beijing 100081, China.\", \"College of Computer and Information Engineering, Xinjiang Agricultural University, Urumqi 830052, China.\", \"Agricultural Information Institute of CAAS, Beijing 100081, China.\", \"Agricultural Information Institute of CAAS, Beijing 100081, China.\", \"National Nanfan Research Institute of CAAS, Sanya, 572024, China.\", \"Agricultural Information Institute of CAAS, Beijing 100081, China.\", \"National Nanfan Research Institute of CAAS, Sanya, 572024, China.\"], \"LA\": [\"eng\"], \"PT\": [\"Journal Article\"], \"DEP\": \"20250909\", \"PL\": \"Netherlands\", \"TA\": \"Data Brief\", \"JT\": \"Data in brief\", \"JID\": \"101654995\", \"PMC\": \"PMC12477916\", \"OTO\": [\"NOTNLM\"], \"OT\": [\"Broiler farming\", \"Extraction of knowledge\", \"Knowledge graph\", \"Large language model\"], \"EDAT\": \"2025/10/01 12:01\", \"MHDA\": \"2025/10/01 12:02\", \"PMCR\": [\"2025/09/09\"], \"CRDT\": [\"2025/10/01 04:58\"], \"PHST\": [\"2025/05/23 00:00 [received]\", \"2025/08/01 00:00 [revised]\", \"2025/08/25 00:00 [accepted]\", \"2025/10/01 12:02 [medline]\", \"2025/10/01 12:01 [pubmed]\", \"2025/10/01 04:58 [entrez]\", \"2025/09/09 00:00 [pmc-release]\"], \"AID\": [\"S2352-3409(25)00740-1 [pii]\", \"112018 [pii]\", \"10.1016/j.dib.2025.112018 [doi]\"], \"PST\": \"epublish\", \"SO\": \"Data Brief. 2025 Sep 9;62:112018. doi: 10.1016/j.dib.2025.112018. eCollection 2025 Oct.\"}, {\"PMID\": \"40968918\", \"OWN\": \"NLM\", \"STAT\": \"PubMed-not-MEDLINE\", \"LR\": \"20250922\", \"IS\": \"1424-8220 (Electronic) 1424-8220 (Linking)\", \"VI\": \"25\", \"IP\": \"11\", \"DP\": \"2025 May 26\", \"TI\": \"Augmenting Orbital Debris Identification with Neo4j-Enabled Graph-Based Retrieval-Augmented Generation for Multimodal Large Language Models.\", \"LID\": \"10.3390/s25113352 [doi] 3352\", \"AB\": \"This preliminary study covers the construction and application of a Graph-based Retrieval-Augmented Generation (GraphRAG) system integrating a multimodal LLM, Large Language and Vision Assistant (LLaVA) with graph database software (Neo4j) to enhance LLM output quality through structured knowledge retrieval. This is aimed at the field of orbital debris detection, proposed to support the current intelligent methods for such detection by introducing the beneficial properties of both LLMs and a corpus of external information. By constructing a dynamic knowledge graph from relevant research papers, context-aware retrieval is enabled, improving factual accuracy and minimizing hallucinations. The system extracts, summarizes, and embeds research papers into a Neo4j graph database, with API-powered LLM-generated relationships enriching interconnections. Querying this graph allows for contextual ranking of relevant documents, which are then provided as context to the LLM through prompt engineering during the inference process. A case study applying the technology to a synthetic image of orbital debris is discussed. Qualitative results indicate that the inclusion of GraphRAG and external information result in successful retrieval of information and reduced hallucinations. Further work to refine the system is necessary, as well as establishing benchmark tests to assess performance quantitatively. This approach offers a scalable and interpretable method for enhanced domain-specific knowledge retrieval, improving the qualitative quality of the LLM's output when tasked with description-based activities.\", \"FAU\": [\"Roll, Daniel S\", \"Kurt, Zeyneb\", \"Li, Yulei\", \"Woo, Wai Lok\"], \"AU\": [\"Roll DS\", \"Kurt Z\", \"Li Y\", \"Woo WL\"], \"AUID\": [\"ORCID: 0009-0002-0927-1598\", \"ORCID: 0000-0003-3186-8091\", \"ORCID: 0000-0003-3579-6179\", \"ORCID: 0000-0002-8698-7605\"], \"AD\": [\"Department of Mathematics, Physics and Electrical Engineering, Faculty of Engineering and Environment, Northumbria University, Newcastle City Campus, College Street, Newcastle-upon-Tyne NE1 8ST, UK.\", \"Information School, The University of Sheffield, The Wave, 2 Whitham Road, Sheffield S10 2AH, UK.\", \"Department of Mathematics, Physics and Electrical Engineering, Faculty of Engineering and Environment, Northumbria University, Newcastle City Campus, College Street, Newcastle-upon-Tyne NE1 8ST, UK.\", \"Department of Mathematics, Physics and Electrical Engineering, Faculty of Engineering and Environment, Northumbria University, Newcastle City Campus, College Street, Newcastle-upon-Tyne NE1 8ST, UK.\"], \"LA\": [\"eng\"], \"GR\": [\"ST/W006790/1/UK Research and Innovation (UKRI) Science and Technology 861\", \"Facilities Council (STF)/\"], \"PT\": [\"Journal Article\"], \"DEP\": \"20250526\", \"PL\": \"Switzerland\", \"TA\": \"Sensors (Basel)\", \"JT\": \"Sensors (Basel, Switzerland)\", \"JID\": \"101204366\", \"SB\": \"IM\", \"PMC\": \"PMC12157151\", \"OTO\": [\"NOTNLM\"], \"OT\": [\"graph databases\", \"knowledge retrieval\", \"large language models\", \"orbital debris\", \"retrieval-augmented generation\", \"space situational awareness\"], \"COIS\": [\"The authors declare no conflicts of interest.\"], \"EDAT\": \"2025/09/19 06:33\", \"MHDA\": \"2025/09/19 06:34\", \"PMCR\": [\"2025/05/26\"], \"CRDT\": [\"2025/09/19 04:36\"], \"PHST\": [\"2025/04/26 00:00 [received]\", \"2025/05/21 00:00 [revised]\", \"2025/05/22 00:00 [accepted]\", \"2025/09/19 06:34 [medline]\", \"2025/09/19 06:33 [pubmed]\", \"2025/09/19 04:36 [entrez]\", \"2025/05/26 00:00 [pmc-release]\"], \"AID\": [\"s25113352 [pii]\", \"sensors-25-03352 [pii]\", \"10.3390/s25113352 [doi]\"], \"PST\": \"epublish\", \"SO\": \"Sensors (Basel). 2025 May 26;25(11):3352. doi: 10.3390/s25113352.\"}, {\"PMID\": \"40804424\", \"OWN\": \"NLM\", \"STAT\": \"MEDLINE\", \"DCOM\": \"20250826\", \"LR\": \"20250826\", \"IS\": \"2041-1480 (Electronic)\", \"VI\": \"16\", \"IP\": \"1\", \"DP\": \"2025 Aug 13\", \"TI\": \"BASIL DB: bioactive semantic integration and linking database.\", \"PG\": \"14\", \"LID\": \"10.1186/s13326-025-00336-3 [doi] 14\", \"AB\": \"BACKGROUND: Bioactive compounds found in foods and plants can provide health benefits, including antioxidant and anti-inflammatory effects. Research into their role in disease prevention and personalized nutrition is expanding, but challenges such as data complexity, inconsistent methods, and the rapid growth of scientific literature can hinder progress. To address these issues, we developed BASIL DB (BioActive Semantic Integration and Linking Database), a knowledge graph (KG) database that leverages natural language processing (NLP) techniques to streamline data organization and analysis. This automated approach offers greater scalability and comprehensiveness than traditional methods such as manual data curation and entry. CONSTRUCTION AND CONTENT: The process of constructing the BASIL DB is divided into four fundamental steps: data collection, data preprocessing, data extraction, and data integration. Data on bioactives and foods are sourced from structured databases. The relevant randomized controlled trials (RCTs) were extracted from PubMed. The data are then prepared by cleaning inconsistencies and structuring them for analysis. In the data extraction phase, NLP tools, including a large language model (LLM), are utilized to analyze clinical trials and extract data on bioactive compounds and their health impacts. The integration phase compiles these data into a knowledge graph, which consists of the entities Foods, Bioactives, and Health Conditions as nodes and their interactions as edges. To quantify the relationships/interactions between these entities, we generate a weight for each edge on the basis of empirical evidence and methodological rigor. UTILITY AND DISCUSSION: The BASIL DB incorporates 433 compounds, 40296 research papers, 7256 health effects, and 4197 food items. The database features query and visualization capabilities, including interactive graphs and custom filtering options, that showcase different aspects of the data. Users are able to explore the relationships between bioactives and health effects, enhancing both research efficiency and insight discovery. CONCLUSION: The BASIL DB is a knowledge graph database of bioactive compounds. This study provides a structured resource for exploring the relationships among bioactives, foods, and health outcomes, representing a step toward a more systematic and data-driven approach to understanding the health effects of bioactive compounds. Future work will focus on expanding the database and refining the utilized methods. Extending the BASIL DB will help bridge the gap between traditional and conventional approaches to nutrition, guiding future research in bioactive compound discovery and health optimization. AVAILABILITY: Users can access and explore the data via https://basil-db.github.io/info.html or fork and run the respective script via https://github.com/basil-db/script .\", \"CI\": [\"(c) 2025. The Author(s).\"], \"FAU\": [\"Jackson, David\", \"Groth, Paul\", \"Harmouch, Hazar\"], \"AU\": [\"Jackson D\", \"Groth P\", \"Harmouch H\"], \"AD\": [\"University of Amsterdam, Amsterdam, The Netherlands. d.i.jackson@uva.nl.\", \"University of Amsterdam, Amsterdam, The Netherlands.\", \"University of Amsterdam, Amsterdam, The Netherlands.\"], \"LA\": [\"eng\"], \"PT\": [\"Journal Article\"], \"DEP\": \"20250813\", \"PL\": \"England\", \"TA\": \"J Biomed Semantics\", \"JT\": \"Journal of biomedical semantics\", \"JID\": \"101531992\", \"SB\": \"IM\", \"MH\": [\"*Semantics\", \"*Natural Language Processing\", \"*Databases, Factual\"], \"PMC\": \"PMC12351831\", \"OTO\": [\"NOTNLM\"], \"OT\": [\"Bioactive compounds\", \"Clinical trials\", \"Data integration\", \"Evidence-based health\", \"Knowledge graph\", \"Natural language processing (NLP)\", \"PubMed\"], \"COIS\": [\"Declarations. Ethics approval and consent to participate: Not applicable, as this\", \"study uses publicly available, de-identified data from existing literature.\", \"Consent for publication: Not applicable. Competing interests: The authors declare\", \"no competing interests.\"], \"EDAT\": \"2025/08/14 00:28\", \"MHDA\": \"2025/08/31 20:07\", \"PMCR\": [\"2025/08/13\"], \"CRDT\": [\"2025/08/13 23:49\"], \"PHST\": [\"2025/03/05 00:00 [received]\", \"2025/08/06 00:00 [accepted]\", \"2025/08/31 20:07 [medline]\", \"2025/08/14 00:28 [pubmed]\", \"2025/08/13 23:49 [entrez]\", \"2025/08/13 00:00 [pmc-release]\"], \"AID\": [\"10.1186/s13326-025-00336-3 [pii]\", \"336 [pii]\", \"10.1186/s13326-025-00336-3 [doi]\"], \"PST\": \"epublish\", \"SO\": \"J Biomed Semantics. 2025 Aug 13;16(1):14. doi: 10.1186/s13326-025-00336-3.\"}, {\"PMID\": \"40700523\", \"OWN\": \"NLM\", \"STAT\": \"MEDLINE\", \"DCOM\": \"20250729\", \"LR\": \"20250805\", \"IS\": \"1532-298X (Electronic) 1040-4651 (Print) 1040-4651 (Linking)\", \"VI\": \"37\", \"IP\": \"7\", \"DP\": \"2025 Jul 1\", \"TI\": \"PlantConnectome: A knowledge graph database encompassing >71,000 plant articles.\", \"LID\": \"10.1093/plcell/koaf169 [doi] koaf169\", \"AB\": \"One of the main quests in plant biology is understanding how gene products and metabolites work together to form complex networks that drive plant development and responses to environmental stimuli. However, the ever-growing volume and diversity of scientific literature make it increasingly challenging to stay current with the latest advances in functional genetics studies. Here, we tackled this challenge by deploying the text-mining capacities of large language models to process over 71,000 plant biology abstracts. Our approach presents nearly 5 million functional relationships between 2.4 million biological entities-genes or gene products, metabolites, tissues, and others-with a high accuracy of over 85%. We encapsulated these findings in the user-friendly database PlantConnectome and demonstrated its diverse utility by providing insights into gene regulatory networks, protein-protein interactions, and stress responses. We believe this innovative use of artificial intelligence (AI) in the life sciences will allow plant scientists to keep up to date with the rapidly growing corpus of scientific literature. PlantConnectome is available at https://plant.connectome.tools/.\", \"CI\": [\"(c) The Author(s) 2025. Published by Oxford University Press on behalf of American\", \"Society of Plant Biologists.\"], \"FAU\": [\"Lim, Shan Chun\", \"Itharajula, Manoj\", \"Moller, Mads Harder\", \"Sunil, Rohan Shawn\", \"Fo, Kevin\", \"Chuah, Yu Song\", \"Foo, Herman\", \"Davey, Emilia Emmanuelle\", \"Fullwood, Melissa\", \"Thibault, Guillaume\", \"Mutwil, Marek\"], \"AU\": [\"Lim SC\", \"Itharajula M\", \"Moller MH\", \"Sunil RS\", \"Fo K\", \"Chuah YS\", \"Foo H\", \"Davey EE\", \"Fullwood M\", \"Thibault G\", \"Mutwil M\"], \"AD\": [\"School of Biological Sciences, Nanyang Technological University, 60 Nanyang Drive, Singapore 637551, Singapore.\", \"School of Biological Sciences, Nanyang Technological University, 60 Nanyang Drive, Singapore 637551, Singapore.\", \"Section of Plant Biochemistry, Department of Plant and Environmental Sciences, University of Copenhagen, Thorvaldsensvej 40, Frederiksberg, Copenhagen 1871, Denmark.\", \"School of Biological Sciences, Nanyang Technological University, 60 Nanyang Drive, Singapore 637551, Singapore.\", \"School of Biological Sciences, Nanyang Technological University, 60 Nanyang Drive, Singapore 637551, Singapore.\", \"School of Biological Sciences, Nanyang Technological University, 60 Nanyang Drive, Singapore 637551, Singapore.\", \"School of Biological Sciences, Nanyang Technological University, 60 Nanyang Drive, Singapore 637551, Singapore.\", \"School of Biological Sciences, Nanyang Technological University, 60 Nanyang Drive, Singapore 637551, Singapore.\", \"School of Biological Sciences, Nanyang Technological University, 60 Nanyang Drive, Singapore 637551, Singapore.\", \"School of Biological Sciences, Nanyang Technological University, 60 Nanyang Drive, Singapore 637551, Singapore.\", \"School of Biological Sciences, Nanyang Technological University, 60 Nanyang Drive, Singapore 637551, Singapore.\", \"Section of Plant Biochemistry, Department of Plant and Environmental Sciences, University of Copenhagen, Thorvaldsensvej 40, Frederiksberg, Copenhagen 1871, Denmark.\"], \"AUID\": [\"ORCID: 0000-0003-0321-7865\", \"ORCID: 0000-0002-7848-0126\"], \"LA\": [\"eng\"], \"GR\": [\"440046237/Novo Nordisk Starting/\", \"NTU/\"], \"PT\": [\"Journal Article\"], \"PL\": \"England\", \"TA\": \"Plant Cell\", \"JT\": \"The Plant cell\", \"JID\": \"9208688\", \"SB\": \"IM\", \"MH\": [\"*Plants/genetics/metabolism\", \"Data Mining\", \"Gene Regulatory Networks\", \"*Databases, Factual\"], \"PMC\": \"PMC12290883\", \"COIS\": [\"Conflict of interest statement. None declared.\"], \"EDAT\": \"2025/07/23 18:28\", \"MHDA\": \"2025/07/29 18:32\", \"PMCR\": [\"2025/07/23\"], \"CRDT\": [\"2025/07/23 14:03\"], \"PHST\": [\"2025/05/01 00:00 [received]\", \"2025/05/10 00:00 [accepted]\", \"2025/07/29 18:32 [medline]\", \"2025/07/23 18:28 [pubmed]\", \"2025/07/23 14:03 [entrez]\", \"2025/07/23 00:00 [pmc-release]\"], \"AID\": [\"8211126 [pii]\", \"koaf169 [pii]\", \"10.1093/plcell/koaf169 [doi]\"], \"PST\": \"ppublish\", \"SO\": \"Plant Cell. 2025 Jul 1;37(7):koaf169. doi: 10.1093/plcell/koaf169.\"}, {\"PMID\": \"40648170\", \"OWN\": \"NLM\", \"STAT\": \"PubMed-not-MEDLINE\", \"LR\": \"20250716\", \"IS\": \"1424-8220 (Electronic) 1424-8220 (Linking)\", \"VI\": \"25\", \"IP\": \"13\", \"DP\": \"2025 Jun 23\", \"TI\": \"Knowledge-Graph-Driven Fault Diagnosis Methods for Intelligent Production Lines.\", \"LID\": \"10.3390/s25133912 [doi] 3912\", \"AB\": \"In order to enhance the management and application of fault knowledge within intelligent production lines, thereby increasing the efficiency of fault diagnosis and ensuring the stable and reliable operation of these systems, we propose a fault diagnosis methodology that leverages knowledge graphs. First, we designed an ontology model for fault knowledge by integrating textual features from various components of the production line with expert insights. Second, we employed the ALBERT-BiLSTM-Attention-CRF model to achieve named entity and relationship recognition for faults in intelligent production lines. The introduction of the ALBERT model resulted in a 7.3% improvement in the F(1) score compared to the BiLSTM-CRF model. Additionally, incorporating the attention mechanism in relationship extraction led to a 7.37% increase in the F(1) score. Finally, we utilized the Neo4j graph database to facilitate the storage and visualization of fault knowledge, validating the effectiveness of our proposed method through a case study on fault diagnosis in CNC machining centers. The research findings indicate that this method excels in recognizing textual entities and relationships related to faults in intelligent production lines, effectively leveraging prior knowledge of faults across various components and elucidating their causes. This approach provides maintenance personnel with an intuitive tool for fault diagnosis and decision support, thereby enhancing diagnostic accuracy and efficiency.\", \"FAU\": [\"Chen, Yanjun\", \"Zhou, Min\", \"Zhang, Meizhou\", \"Zha, Meng\"], \"AU\": [\"Chen Y\", \"Zhou M\", \"Zhang M\", \"Zha M\"], \"AUID\": [\"ORCID: 0009-0004-9321-5525\"], \"AD\": [\"Key Laboratory of Metallurgical Equipment and Control Technology, Ministry of Education, Wuhan University of Science and Technology, Wuhan 430081, China.\", \"Hubei Key Laboratory of Mechanical Transmission and Manufacturing Engineering, Wuhan University of Science and Technology, Wuhan 430081, China.\", \"Institute of Precision Manufacturing, Wuhan University of Science and Technology, Wuhan 430081, China.\", \"Key Laboratory of Metallurgical Equipment and Control Technology, Ministry of Education, Wuhan University of Science and Technology, Wuhan 430081, China.\", \"Hubei Key Laboratory of Mechanical Transmission and Manufacturing Engineering, Wuhan University of Science and Technology, Wuhan 430081, China.\", \"Institute of Precision Manufacturing, Wuhan University of Science and Technology, Wuhan 430081, China.\", \"Key Laboratory of Metallurgical Equipment and Control Technology, Ministry of Education, Wuhan University of Science and Technology, Wuhan 430081, China.\", \"Hubei Key Laboratory of Mechanical Transmission and Manufacturing Engineering, Wuhan University of Science and Technology, Wuhan 430081, China.\", \"Institute of Precision Manufacturing, Wuhan University of Science and Technology, Wuhan 430081, China.\", \"Key Laboratory of Metallurgical Equipment and Control Technology, Ministry of Education, Wuhan University of Science and Technology, Wuhan 430081, China.\", \"Hubei Key Laboratory of Mechanical Transmission and Manufacturing Engineering, Wuhan University of Science and Technology, Wuhan 430081, China.\", \"Institute of Precision Manufacturing, Wuhan University of Science and Technology, Wuhan 430081, China.\"], \"LA\": [\"eng\"], \"GR\": [\"51975431/The National Natural Science Foundation of China/\"], \"PT\": [\"Journal Article\"], \"DEP\": \"20250623\", \"PL\": \"Switzerland\", \"TA\": \"Sensors (Basel)\", \"JT\": \"Sensors (Basel, Switzerland)\", \"JID\": \"101204366\", \"SB\": \"IM\", \"PMC\": \"PMC12252116\", \"OTO\": [\"NOTNLM\"], \"OT\": [\"BiLSTM (bidirectional long short-term memory network)\", \"entity recognition\", \"fault diagnosis\", \"intelligent production line\", \"knowledge graph\"], \"COIS\": [\"The authors have thoroughly examined their potential conflicts of interest and\", \"declare none, ensuring the integrity and impartiality of their research.\"], \"EDAT\": \"2025/07/13 03:36\", \"MHDA\": \"2025/07/13 03:37\", \"PMCR\": [\"2025/06/23\"], \"CRDT\": [\"2025/07/12 01:11\"], \"PHST\": [\"2025/04/14 00:00 [received]\", \"2025/06/18 00:00 [revised]\", \"2025/06/20 00:00 [accepted]\", \"2025/07/13 03:37 [medline]\", \"2025/07/13 03:36 [pubmed]\", \"2025/07/12 01:11 [entrez]\", \"2025/06/23 00:00 [pmc-release]\"], \"AID\": [\"s25133912 [pii]\", \"sensors-25-03912 [pii]\", \"10.3390/s25133912 [doi]\"], \"PST\": \"epublish\", \"SO\": \"Sensors (Basel). 2025 Jun 23;25(13):3912. doi: 10.3390/s25133912.\"}, {\"PMID\": \"40640288\", \"OWN\": \"NLM\", \"STAT\": \"PubMed-not-MEDLINE\", \"LR\": \"20250713\", \"IS\": \"2045-2322 (Electronic) 2045-2322 (Linking)\", \"VI\": \"15\", \"IP\": \"1\", \"DP\": \"2025 Jul 11\", \"TI\": \"Construction of knowledge graph for gas polyethylene pipelines based on ALBERT-BiGRU-CRF.\", \"PG\": \"25002\", \"LID\": \"10.1038/s41598-025-08447-8 [doi] 25002\", \"AB\": \"With the advancement of intelligent operation and maintenance (O&M) for gas polyethylene pipelines, effectively managing and utilizing pipeline data has become crucial for the development of information technology. However, the current O&M process faces challenges, including weak information relevance and low decision-making efficiency. In contrast to the well-established O&M knowledge system for metal pipelines, failure modes in polyethylene pipelines-such as material ageing and third-party damage-lack systematic knowledge modelling. Additionally, the multidimensional correlations among key entities (e.g., pipe segments, welded joints, and cathodic protection devices) remain unexplored. To address this, this paper proposes a knowledge graph construction method based on ALBERT-BiGRU-CRF for gas polyethylene pipeline operation and maintenance. First, structured, unstructured and semi-structured gas polyethylene pipeline operation and maintenance data were preprocessed and integrated to construct a high-quality 3D training data set containing 18 different parameter types. Next, the improved ALBERT, incorporating features from various levels, generates high-quality word vectors, which serve as input feature vectors for the bidirectional gated recurrent unit (BiGRU) model to improve the model's semantic comprehension ability in processing pipeline O&M data. Additionally, the CRF layer captures dependencies within entity sequences and optimizes label prediction results, addressing the sequence dependency issue in entity recognition. Finally, the model extracts knowledge to form a ternary structure, which is imported into the Neo4j graph database for storage and visualization. Experimental results demonstrate that the ALBERT-BiGRU-CRF model performs well in recognizing the dataset, with accuracy, recall, and F1 scores reaching 95.35, 96.51, and 95.93%, respectively. In comparison with other models, the accuracy, recall, and F1 scores show significant improvements.\", \"CI\": [\"(c) 2025. The Author(s).\"], \"FAU\": [\"Yu, Zhilong\", \"Zhou, Juan\", \"Wang, Qiang\", \"Zhou, Haiting\", \"Song, Yun\", \"Zong, Chenjia\"], \"AU\": [\"Yu Z\", \"Zhou J\", \"Wang Q\", \"Zhou H\", \"Song Y\", \"Zong C\"], \"AD\": [\"College of Quality and Standardization, China Jiliang University, Hangzhou, 310018, China.\", \"College of Quality and Standardization, China Jiliang University, Hangzhou, 310018, China. zhoujuan@cjlu.edu.cn.\", \"College of Energy Environment and Safety Engineering and College of Carbon Metrology, China Jiliang University, Hangzhou, 310018, China.\", \"College of Quality and Standardization, China Jiliang University, Hangzhou, 310018, China.\", \"College of Quality and Standardization, China Jiliang University, Hangzhou, 310018, China.\", \"College of Quality and Standardization, China Jiliang University, Hangzhou, 310018, China.\"], \"LA\": [\"eng\"], \"GR\": [\"No. 2023YFF0611600/The National Key Research and Development Program of China/\"], \"PT\": [\"Journal Article\"], \"DEP\": \"20250711\", \"PL\": \"England\", \"TA\": \"Sci Rep\", \"JT\": \"Scientific reports\", \"JID\": \"101563288\", \"SB\": \"IM\", \"PMC\": \"PMC12246125\", \"OTO\": [\"NOTNLM\"], \"OT\": [\"Gas polyethylene pipes\", \"Knowledge extraction\", \"Knowledge graph\", \"Named entity recognition\"], \"COIS\": [\"Declarations. Competing interests: The authors declare no competing interests.\"], \"EDAT\": \"2025/07/11 00:26\", \"MHDA\": \"2025/07/11 00:27\", \"PMCR\": [\"2025/07/11\"], \"CRDT\": [\"2025/07/10 23:25\"], \"PHST\": [\"2025/02/28 00:00 [received]\", \"2025/06/20 00:00 [accepted]\", \"2025/07/11 00:27 [medline]\", \"2025/07/11 00:26 [pubmed]\", \"2025/07/10 23:25 [entrez]\", \"2025/07/11 00:00 [pmc-release]\"], \"AID\": [\"10.1038/s41598-025-08447-8 [pii]\", \"8447 [pii]\", \"10.1038/s41598-025-08447-8 [doi]\"], \"PST\": \"epublish\", \"SO\": \"Sci Rep. 2025 Jul 11;15(1):25002. doi: 10.1038/s41598-025-08447-8.\"}, {\"PMID\": \"40580451\", \"OWN\": \"NLM\", \"STAT\": \"MEDLINE\", \"DCOM\": \"20250717\", \"LR\": \"20250719\", \"IS\": \"1367-4811 (Electronic) 1367-4803 (Print) 1367-4803 (Linking)\", \"VI\": \"41\", \"IP\": \"7\", \"DP\": \"2025 Jul 1\", \"TI\": \"KGG: a fully automated workflow for creating disease-specific knowledge graphs.\", \"LID\": \"10.1093/bioinformatics/btaf383 [doi] btaf383\", \"AB\": \"MOTIVATION: Knowledge graphs (KGs) in life sciences have become an important application of systems biology as they delineate complex biological and pathophysiological phenomena. They are composed of biological and chemical entities represented with standard ontologies to comply with Findable, Accessible, Interoperable and Reusable (FAIR) principles. Alongside serving as a graph database, KGs hold the potential to address complex scientific queries and facilitate downstream analyses. However, the process of constructing KGs is expensive and time consuming as it primarily relies on manual curation from published literature and experimental data. The existing text-mining workflows are still in their infancy and fail to achieve the accuracy and reliability of manual curation. RESULTS: Knowledge graph generator (KGG) is an automated workflow for representing chemotype and phenotype of diseases and medical conditions. It embeds the underlying schema of curated databases such as OpenTargets, Uniprot, ChEMBL, Integrated Interactions Database and GWAS Central resembling a clockwork-esque mechanism. The resultant KG is a comprehensive and rational assembly of disease-associated entities such as proteins, protein-related pathways, biological processes and functions, genetic variants, chemicals, mechanism of actions, assays and adverse effects. As use cases, we have used KGs to identify shared entities for possible link of comorbidity and compared them with KGs from other sources. We have also demonstrated a use case of identifying putative new targets and repurposing drug candidates in Parkinson's Disease. Lastly, we have developed reusable workflows to explore drug-likeness of chemicals and identify structures of proteins. AVAILABILITY AND IMPLEMENTATION: The resources and codes for KGG are publicly available at: https://github.com/Fraunhofer-ITMP/kgg.\", \"CI\": [\"(c) The Author(s) 2025. Published by Oxford University Press.\"], \"FAU\": [\"Karki, Reagon\", \"Gadiya, Yojana\", \"Zaliani, Andrea\", \"Pokharel, Bishab\", \"Babaiha, Negin Sadat\", \"Ostaszewski, Marek\", \"Hofmann-Apitius, Martin\", \"Gribbon, Philip\"], \"AU\": [\"Karki R\", \"Gadiya Y\", \"Zaliani A\", \"Pokharel B\", \"Babaiha NS\", \"Ostaszewski M\", \"Hofmann-Apitius M\", \"Gribbon P\"], \"AUID\": [\"ORCID: 0000-0002-1815-0037\", \"ORCID: 0000-0002-7683-0452\", \"ORCID: 0000-0003-1473-370X\", \"ORCID: 0000-0001-9012-6720\"], \"AD\": [\"Fraunhofer Institute for Translational Medicine and Pharmacology (ITMP), Hamburg, 22525, Germany.\", \"Fraunhofer Cluster of Excellence for Immune-Mediated Diseases (CIMD), Frankfurt, 60590, Germany.\", \"Fraunhofer Institute for Translational Medicine and Pharmacology (ITMP), Hamburg, 22525, Germany.\", \"Fraunhofer Cluster of Excellence for Immune-Mediated Diseases (CIMD), Frankfurt, 60590, Germany.\", \"Bonn-Aachen International Center for Information Technology (B-IT), University of Bonn, Bonn, 53113, Germany.\", \"Fraunhofer Institute for Translational Medicine and Pharmacology (ITMP), Hamburg, 22525, Germany.\", \"Fraunhofer Cluster of Excellence for Immune-Mediated Diseases (CIMD), Frankfurt, 60590, Germany.\", \"Fraunhofer Institute for Translational Medicine and Pharmacology (ITMP), Hamburg, 22525, Germany.\", \"Fraunhofer Cluster of Excellence for Immune-Mediated Diseases (CIMD), Frankfurt, 60590, Germany.\", \"University of Hamburg, Department of Informatics, Hamburg, 22527, Germany.\", \"Bonn-Aachen International Center for Information Technology (B-IT), University of Bonn, Bonn, 53113, Germany.\", \"Department of Bioinformatics, Fraunhofer Institute for Algorithms and Scientific Computing (SCAI), Sankt Augustin, 53757, Germany.\", \"Luxembourg Centre for Systems Biomedicine (LCSB), University of Luxembourg, Belvaux, L-4367, Luxembourg.\", \"Bonn-Aachen International Center for Information Technology (B-IT), University of Bonn, Bonn, 53113, Germany.\", \"Department of Bioinformatics, Fraunhofer Institute for Algorithms and Scientific Computing (SCAI), Sankt Augustin, 53757, Germany.\", \"Fraunhofer Institute for Translational Medicine and Pharmacology (ITMP), Hamburg, 22525, Germany.\", \"Fraunhofer Cluster of Excellence for Immune-Mediated Diseases (CIMD), Frankfurt, 60590, Germany.\"], \"LA\": [\"eng\"], \"GR\": [\"101046203/Horizon Europe's BY-COVID/\", \"101017536/EOSC Future Project/\"], \"PT\": [\"Journal Article\"], \"PL\": \"England\", \"TA\": \"Bioinformatics\", \"JT\": \"Bioinformatics (Oxford, England)\", \"JID\": \"9808944\", \"SB\": \"IM\", \"MH\": [\"Workflow\", \"Humans\", \"*Software\", \"Data Mining/methods\", \"Databases, Factual\", \"*Computational Biology/methods\", \"*Disease\", \"*Systems Biology/methods\"], \"PMC\": \"PMC12270262\", \"EDAT\": \"2025/07/01 04:50\", \"MHDA\": \"2025/07/17 18:26\", \"PMCR\": [\"2025/06/28\"], \"CRDT\": [\"2025/06/28 12:23\"], \"PHST\": [\"2024/12/24 00:00 [received]\", \"2025/05/29 00:00 [revised]\", \"2025/06/27 00:00 [accepted]\", \"2025/07/17 18:26 [medline]\", \"2025/07/01 04:50 [pubmed]\", \"2025/06/28 12:23 [entrez]\", \"2025/06/28 00:00 [pmc-release]\"], \"AID\": [\"8177146 [pii]\", \"btaf383 [pii]\", \"10.1093/bioinformatics/btaf383 [doi]\"], \"PST\": \"ppublish\", \"SO\": \"Bioinformatics. 2025 Jul 1;41(7):btaf383. doi: 10.1093/bioinformatics/btaf383.\"}, {\"PMID\": \"40580448\", \"OWN\": \"NLM\", \"STAT\": \"MEDLINE\", \"DCOM\": \"20250713\", \"LR\": \"20250921\", \"IS\": \"1367-4811 (Electronic) 1367-4803 (Print) 1367-4803 (Linking)\", \"VI\": \"41\", \"IP\": \"7\", \"DP\": \"2025 Jul 1\", \"TI\": \"PloverDB: a high-performance platform for serving biomedical knowledge graphs as standards-compliant web APIs.\", \"LID\": \"10.1093/bioinformatics/btaf380 [doi] btaf380\", \"AB\": \"SUMMARY: Knowledge graphs are increasingly being used to integrate heterogeneous biomedical knowledge and data. General-purpose graph database management systems such as Neo4j are often used to host and search knowledge graphs, but such tools come with overhead and leave biomedical-specific standards compliance and reasoning to the user. Interoperability across biomedical knowledge bases and reasoning systems necessitates the use of standards such as those adopted by the Biomedical Data Translator consortium. We present PloverDB, a comprehensive software platform for hosting and efficiently serving biomedical knowledge graphs as standards-compliant web application programming interfaces. In addition to fundamental back-end knowledge reasoning tasks, PloverDB automatically handles entity resolution, exposure of standardized metadata and test data, and multiplexing of knowledge graphs, all in a single platform designed specifically for efficient query answering and ease of deployment. PloverDB increases data accessibility and utility by allowing data providers to quickly serve their biomedical knowledge graphs as standards-compliant web services. AVAILABILITY AND IMPLEMENTATION: PloverDB's source code and technical documentation are publicly available under an MIT License at github:RTXteam/PloverDB, archived on Zenodo at doi:10.5281/zenodo.15454600.\", \"CI\": [\"(c) The Author(s) 2025. Published by Oxford University Press.\"], \"FAU\": [\"Glen, Amy K\", \"Deutsch, Eric W\", \"Ramsey, Stephen A\"], \"AU\": [\"Glen AK\", \"Deutsch EW\", \"Ramsey SA\"], \"AUID\": [\"ORCID: 0000-0003-3691-2290\", \"ORCID: 0000-0001-8732-0928\", \"ORCID: 0000-0002-2168-5403\"], \"AD\": [\"School of Electrical Engineering and Computer Science, Oregon State University, Corvallis, OR 97331, United States.\", \"Institute for Systems Biology, Seattle, WA 98109, United States.\", \"School of Electrical Engineering and Computer Science, Oregon State University, Corvallis, OR 97331, United States.\", \"Department of Biomedical Sciences, Oregon State University, Corvallis, OR 97331, United States.\"], \"LA\": [\"eng\"], \"GR\": [\"OT2TR002520/TR/NCATS NIH HHS/United States\", \"OT2 TR002520/TR/NCATS NIH HHS/United States\", \"OT2 TR003428/TR/NCATS NIH HHS/United States\", \"NH/NIH HHS/United States\", \"OT2TR003428/TR/NCATS NIH HHS/United States\"], \"PT\": [\"Journal Article\"], \"PL\": \"England\", \"TA\": \"Bioinformatics\", \"JT\": \"Bioinformatics (Oxford, England)\", \"JID\": \"9808944\", \"SB\": \"IM\", \"UOF\": [\"bioRxiv. 2025 May 29:2025.03.09.642156. doi: 10.1101/2025.03.09.642156. PMID:\", \"40161641\"], \"MH\": [\"*Software\", \"*Internet\", \"User-Computer Interface\", \"*Computational Biology/methods\", \"*Computer Graphics\", \"*Database Management Systems\", \"Databases, Factual\"], \"PMC\": \"PMC12255876\", \"EDAT\": \"2025/07/01 04:50\", \"MHDA\": \"2025/07/13 13:18\", \"PMCR\": [\"2025/06/28\"], \"CRDT\": [\"2025/06/28 12:22\"], \"PHST\": [\"2025/03/08 00:00 [received]\", \"2025/05/29 00:00 [revised]\", \"2025/06/26 00:00 [accepted]\", \"2025/07/13 13:18 [medline]\", \"2025/07/01 04:50 [pubmed]\", \"2025/06/28 12:22 [entrez]\", \"2025/06/28 00:00 [pmc-release]\"], \"AID\": [\"8177145 [pii]\", \"btaf380 [pii]\", \"10.1093/bioinformatics/btaf380 [doi]\"], \"PST\": \"ppublish\", \"SO\": \"Bioinformatics. 2025 Jul 1;41(7):btaf380. doi: 10.1093/bioinformatics/btaf380.\"}, {\"PMID\": \"40518789\", \"OWN\": \"NLM\", \"STAT\": \"MEDLINE\", \"DCOM\": \"20250616\", \"LR\": \"20250616\", \"IS\": \"0255-2930 (Print) 0255-2930 (Linking)\", \"VI\": \"45\", \"IP\": \"6\", \"DP\": \"2025 Jun 12\", \"TI\": \"[Rules of moxibustion for low back pain by ZHOU Meisheng based on data mining and knowledge graph technology].\", \"PG\": \"823-833\", \"LID\": \"10.13703/j.0255-2930.20240418-k0004 [doi]\", \"AB\": \"OBJECTIVE: To analyze the rules of moxibustion for low back pain by ZHOU Meisheng by using data mining and knowledge graph technology. METHODS: Taking the medical cases of moxibustion for low back pain from ZHOU Meisheng's legacy manuscripts and existing works as the research objects, information on disease types, symptoms, tongue manifestations, pulse conditions, syndrome patterns, moxibustion methods and acupoints were collected. Frequency statistics and community analysis were conducted by the ancient and modern medical record cloud platform V 2.3.7, cluster analysis of high-frequency acupoints was performed by SPSS26.0, association rule analysis of high-frequency acupoints was performed by SPSS Modeler 18.0, and the generated linked data were imported into Cytoscape 3.9.1 for complex network analysis. Knowledge graph of moxibustion for low back pain by ZHOU Meisheng was constructed based on the results of data mining. The data storage and display of knowledge graph were realized through the Neo4j 3.5.25 graph database, and the Cypher query language was used for knowledge graph retrieval and discovery. RESULTS: A total of 219 medical cases were collected, involving 14 disease types, 85 related clinical symptoms, 5 related TCM syndrome types, and 6 moxibustion methods. The acupoints were mostly attributed to the governor vessel, the bladder meridian of foot-taiyang, non-meridian and non-acupoint areas. The core prescription of acupoints derived from complex network analysis included tender points, Shenshu (BL23), Jinsuo (GV8), Yinjiao (CV7), Yaoyangguan (GV3), Yanglingquan (GB34), which were largely coincides with high-frequency acupoints. Cluster analysis obtained 4 cluster combinations. Community analysis yielded 6 communities, each corresponding to different acupoints.The constructed knowledge graph contained 187 nodes and 696 relationships, by retrieving clinical elements related to low back pain, the disease-moxibustion association graph, disease-acupoint association graph, accompanying symptom-acupoint association graph and syndrome type-matching point association graph were obtained. CONCLUSION: When treating low back pain with moxibustion, ZHOU Meisheng adopts the principle of promoting circulation, distinguishing diseases to determine the treatment, selecting acupoints according to the diseases, and matching points according to the symptoms.And taking tender points, Shenshu (BL23), Jinsuo (GV8), Yinjiao (CV7), Yaoyangguan (GV3), Yanglingquan (GB34) as core acupoints, combined with tenderness point selection, acupoint selection based on meridian and zangfu syndrome differentiation, \\\"sunshine area\\\" acupoint selection, yin-yang acupoint matching. Additionally, he skillfully employs special points, such as Zhongzhu (KI15) and ear tips, pays attention to the reform of moxibustion tools, and innovates the moxibustion techniques, using distinctive moxibustion tools and methods to treat low back pain.\", \"FAU\": [\"Wang, Chi\", \"Zhu, Caifeng\", \"Zhang, Jiayu\", \"Zhou, Bingyuan\", \"Chen, Xiaoyu\", \"Cheng, Le\", \"Xie, Miaomiao\", \"Ding, Xuechun\"], \"AU\": [\"Wang C\", \"Zhu C\", \"Zhang J\", \"Zhou B\", \"Chen X\", \"Cheng L\", \"Xie M\", \"Ding X\"], \"AD\": [\"Second School of Clinical Medicine, Anhui University of CM, Hefei 230061, China.\", \"Third Department of Geriatrics, Second Affiliated Hospital of Anhui University of CM, Hefei 230061.\", \"Second School of Clinical Medicine, Anhui University of CM, Hefei 230061, China.\", \"Second School of Clinical Medicine, Anhui University of CM, Hefei 230061, China.\", \"Second School of Clinical Medicine, Anhui University of CM, Hefei 230061, China.\", \"Second School of Clinical Medicine, Anhui University of CM, Hefei 230061, China.\", \"Second School of Clinical Medicine, Anhui University of CM, Hefei 230061, China.\", \"Second School of Clinical Medicine, Anhui University of CM, Hefei 230061, China.\"], \"LA\": [\"chi\"], \"PT\": [\"English Abstract\", \"Historical Article\", \"Journal Article\"], \"DEP\": \"20250219\", \"PL\": \"China\", \"TA\": \"Zhongguo Zhen Jiu\", \"JT\": \"Zhongguo zhen jiu = Chinese acupuncture & moxibustion\", \"JID\": \"8600658\", \"SB\": \"IM\", \"MH\": [\"*Moxibustion/history/methods\", \"Humans\", \"*Data Mining\", \"*Low Back Pain/therapy/history\", \"Acupuncture Points\", \"History, Ancient\", \"Female\", \"China\", \"Male\", \"Adult\", \"Middle Aged\"], \"OTO\": [\"NOTNLM\"], \"OT\": [\"ZHOU Meisheng\", \"data mining\", \"knowledge graph\", \"low back pain\", \"moxibustion method\"], \"EDAT\": \"2025/06/16 11:11\", \"MHDA\": \"2025/06/16 11:12\", \"CRDT\": [\"2025/06/16 03:03\"], \"PHST\": [\"2024/04/18 00:00 [received]\", \"2025/06/16 11:12 [medline]\", \"2025/06/16 11:11 [pubmed]\", \"2025/06/16 03:03 [entrez]\"], \"AID\": [\"10.13703/j.0255-2930.20240418-k0004 [doi]\"], \"PST\": \"ppublish\", \"SO\": \"Zhongguo Zhen Jiu. 2025 Jun 12;45(6):823-833. doi: 10.13703/j.0255-2930.20240418-k0004. Epub 2025 Feb 19.\"}, {\"PMID\": \"40357433\", \"OWN\": \"NLM\", \"STAT\": \"PubMed-not-MEDLINE\", \"LR\": \"20250514\", \"IS\": \"2297-055X (Print) 2297-055X (Electronic) 2297-055X (Linking)\", \"VI\": \"12\", \"DP\": \"2025\", \"TI\": \"From knowledge silos to integrated insights: building a cardiovascular medication knowledge graph for enhanced medication knowledge retrieval, relationship discovery, and reasoning.\", \"PG\": \"1526247\", \"LID\": \"10.3389/fcvm.2025.1526247 [doi] 1526247\", \"AB\": \"BACKGROUND: Cardiovascular diseases are diverse, intersecting, and characterized by multistage complexity. The growing demand for personalized diagnosis and treatment poses significant challenges to clinical diagnosis and pharmacotherapy, increasing potential medication risks for doctors and patients. The Cardiovascular Medication Guide (CMG) demonstrates distinct advantages in managing cardiovascular disease, serving as a critical reference for front-line doctors in prescription selection and treatment planning. However, most medical knowledge remains fragmented within written records, such as medical files, without a cohesive organizational structure, leading to an absence of clinical support from visualized expert knowledge systems. PURPOSE: This study aims to construct a comprehensive Expert Knowledge Graph of Cardiovascular Medication Guidelines (EKG-CMG) by integrating unstructured and semi-structured Cardiovascular Medication Knowledge (CMK), including clinical guidelines and expert consensus, to create a visually integrated cardiovascular expert knowledge system. METHODS: This study utilized consensus and guidelines from cardiovascular experts to organize and manage structured knowledge. BERT and knowledge extraction techniques capture drug attribute relationships, leading to the construction of the EKG-CMG with fine-grained information. The Neo4j graph database stores expert knowledge, visualizes knowledge structures and semantic relationships, and supports retrieval, discovery, and reasoning of knowledge about medication. A hierarchical-weighted, multidimensional relational model to mine medication relationships through reverse reasoning. Experts participated in an iterative review process, allowing targeted refinement of expert medication knowledge reasoning. RESULTS: We construct an ontology encompassing 12 cardiovascular \\\"medication types\\\" and their \\\"attributes of medication types\\\". Approximately 15,000 entity-relationships include 22,475 medication entities, 2,027 entity categories, and 3,304 relationships. Taking beta-blockers (beta) as an example demonstrates the complete process of ontology to knowledge graph construction and application, encompassing 41 AMTs, 1,197 entity nodes, and 1,351 relationships. The EKG-CMG can complete knowledge retrieval and discovery linked to \\\"one drug for multiple uses,\\\" \\\"combination therapy,\\\" and \\\"precision medication.\\\" Additionally, we analyzed the knowledge reasoning case of cross-symptoms and complex medication for complications. CONCLUSION: The EKG-CMG systematically organizes CMK, effectively addressing the \\\"knowledge island\\\" issues between diseases and drugs. Knowledge potential relationships have been exposed by leveraging EKG-CMG visualization technology, which can facilitate medication semantic retrieval and the exploration and reasoning of complex knowledge relationships.\", \"CI\": [\"(c) 2025 Cui, Zhu, Zhang, Piao and Peng.\"], \"FAU\": [\"Cui, Hongzhen\", \"Zhu, Xiaoyue\", \"Zhang, Wei\", \"Piao, Meihua\", \"Peng, Yunfeng\"], \"AU\": [\"Cui H\", \"Zhu X\", \"Zhang W\", \"Piao M\", \"Peng Y\"], \"AD\": [\"School of Computer and Communication Engineering, University of Science and Technology Beijing, Beijing, China.\", \"School of Computer and Communication Engineering, University of Science and Technology Beijing, Beijing, China.\", \"Key Laboratory of Computing Power Network and Information Security, Ministry of Education, Shandong Computer Science Center (National Supercomputer Center in Jinan), Qilu University of Technology (Shandong Academy of Sciences), Jinan, China.\", \"Shandong Provincial Key Laboratory of Computing Power Internet and Service Computing, Shandong Fundamental Research Center for Computer Science, Jinan, China.\", \"School of Nursing, Chinese Academy of Medical Sciences, Peking Union Medical College, Beijing, China.\", \"School of Computer and Communication Engineering, University of Science and Technology Beijing, Beijing, China.\"], \"LA\": [\"eng\"], \"PT\": [\"Journal Article\"], \"DEP\": \"20250428\", \"PL\": \"Switzerland\", \"TA\": \"Front Cardiovasc Med\", \"JT\": \"Frontiers in cardiovascular medicine\", \"JID\": \"101653388\", \"PMC\": \"PMC12066555\", \"OTO\": [\"NOTNLM\"], \"OT\": [\"EKG-CMG\", \"cardiovascular medication knowledge\", \"expert knowledge graph\", \"knowledge discover and reasoning\", \"multidimensional relationships\", \"ontology construction\"], \"COIS\": [\"The authors declare that the research was conducted in the absence of any\", \"commercial or financial relationships that could be construed as a potential\", \"conflict of interest.\"], \"EDAT\": \"2025/05/13 06:33\", \"MHDA\": \"2025/05/13 06:34\", \"PMCR\": [\"2025/01/01\"], \"CRDT\": [\"2025/05/13 04:55\"], \"PHST\": [\"2024/11/14 00:00 [received]\", \"2025/04/09 00:00 [accepted]\", \"2025/05/13 06:34 [medline]\", \"2025/05/13 06:33 [pubmed]\", \"2025/05/13 04:55 [entrez]\", \"2025/01/01 00:00 [pmc-release]\"], \"AID\": [\"10.3389/fcvm.2025.1526247 [doi]\"], \"PST\": \"epublish\", \"SO\": \"Front Cardiovasc Med. 2025 Apr 28;12:1526247. doi: 10.3389/fcvm.2025.1526247. eCollection 2025.\"}, {\"PMID\": \"40292999\", \"OWN\": \"NLM\", \"STAT\": \"PubMed-not-MEDLINE\", \"LR\": \"20250430\", \"IS\": \"1424-8220 (Electronic) 1424-8220 (Linking)\", \"VI\": \"25\", \"IP\": \"6\", \"DP\": \"2025 Mar 18\", \"TI\": \"Privacy-Preserving Multi-User Graph Intersection Scheme for Wireless Communications in Cloud-Assisted Internet of Things.\", \"LID\": \"10.3390/s25061892 [doi] 1892\", \"AB\": \"Cloud-assisted Internet of Things (IoT) has become the core infrastructure of smart society since it solves the computational power, storage, and collaboration bottlenecks of traditional IoT through resource decoupling and capability complementarity. The development of a graph database and cloud-assisted IoT promotes the research of privacy preserving graph computation. We propose a secure graph intersection scheme that supports multi-user intersection queries in cloud-assisted IoT in this article. The existing work on graph encryption for intersection queries is designed for a single user, which will bring high computational and communication costs for data owners, or cause the risk of secret key leaking if directly applied to multi-user scenarios. To solve these problems, we employ the proxy re-encryption (PRE) that transforms the encrypted graph data with a re-encryption key to enable the graph intersection results to be decrypted by an authorized IoT user using their own private key, while data owners only encrypt their graph data on IoT devices once. In our scheme, different IoT users can query for the intersection of graphs flexibly, while data owners do not need to perform encryption operations every time an IoT user makes a query. Theoretical analysis and simulation results demonstrate that the graph intersection scheme in this paper is secure and practical.\", \"FAU\": [\"Yang, Shumei\"], \"AU\": [\"Yang S\"], \"AD\": [\"School of Computer Science and Engineering, South China University of Technology, Guangzhou 510006, China.\"], \"LA\": [\"eng\"], \"PT\": [\"Journal Article\"], \"DEP\": \"20250318\", \"PL\": \"Switzerland\", \"TA\": \"Sensors (Basel)\", \"JT\": \"Sensors (Basel, Switzerland)\", \"JID\": \"101204366\", \"SB\": \"IM\", \"PMC\": \"PMC11946860\", \"OTO\": [\"NOTNLM\"], \"OT\": [\"Internet of Things (IoT)\", \"cloud computing\", \"graph encryption\", \"graph intersection\", \"privacy preserving\", \"wireless communications\"], \"COIS\": [\"The authors declare no conflicts of interest.\"], \"EDAT\": \"2025/04/28 12:30\", \"MHDA\": \"2025/04/28 12:31\", \"PMCR\": [\"2025/03/18\"], \"CRDT\": [\"2025/04/28 07:33\"], \"PHST\": [\"2025/03/01 00:00 [received]\", \"2025/03/15 00:00 [revised]\", \"2025/03/17 00:00 [accepted]\", \"2025/04/28 12:31 [medline]\", \"2025/04/28 12:30 [pubmed]\", \"2025/04/28 07:33 [entrez]\", \"2025/03/18 00:00 [pmc-release]\"], \"AID\": [\"s25061892 [pii]\", \"sensors-25-01892 [pii]\", \"10.3390/s25061892 [doi]\"], \"PST\": \"epublish\", \"SO\": \"Sensors (Basel). 2025 Mar 18;25(6):1892. doi: 10.3390/s25061892.\"}, {\"PMID\": \"40193601\", \"OWN\": \"NLM\", \"STAT\": \"MEDLINE\", \"DCOM\": \"20250516\", \"LR\": \"20251008\", \"IS\": \"1091-6490 (Electronic) 0027-8424 (Print) 0027-8424 (Linking)\", \"VI\": \"122\", \"IP\": \"15\", \"DP\": \"2025 Apr 15\", \"TI\": \"An integrated AI knowledge graph framework of bacterial enzymology and metabolism.\", \"PG\": \"e2425048122\", \"LID\": \"10.1073/pnas.2425048122 [doi] e2425048122\", \"AB\": \"The study of bacterial metabolism holds immense significance for improving human health and advancing agricultural practices. The prospective applications of genomically encoded bacterial metabolism present a compelling opportunity, particularly in the light of the rapid expansion of genomic sequencing data. Current metabolic inference tools face challenges in scaling with large datasets, leading to increased computational demands, and often exhibit limited inter-relatability and interoperability. Here, we introduce the Integrated Biosynthetic Inference Suite (IBIS), which employs deep learning models and a knowledge graph to facilitate rapid, scalable bacterial metabolic inference. This system leverages a series of Transformer based models to generate high quality, meaningful embeddings for individual enzymes, biosynthetic domains, and metabolic pathways. These embedded representations enable rapid, large-scale comparisons of metabolic proteins and pathways, surpassing the capabilities of conventional methodologies. The examination of evolutionary and functionally conserved metabolites across diverse bacterial species is facilitated by integrating the predictive capabilities of IBIS into a graph database enriched with comprehensive metadata. The consideration of both primary and specialized metabolism, combined with an embedding logic for enzyme discovery, uniquely positions IBIS to identify potential novel metabolic pathways. With the expansion of genomic data necessitating transformative approaches to advance molecular metabolism research, IBIS delivers an AI-driven holistic investigation of bacterial metabolism.\", \"FAU\": [\"Spencer, Norman R\", \"Gunabalasingam, Mathusan\", \"Dial, Keshav\", \"Di, Xiaxia\", \"Malcolm, Tonya\", \"Magarvey, Nathan A\"], \"AU\": [\"Spencer NR\", \"Gunabalasingam M\", \"Dial K\", \"Di X\", \"Malcolm T\", \"Magarvey NA\"], \"AUID\": [\"ORCID: 0000-0003-3405-1079\", \"ORCID: 0009-0001-9171-4118\", \"ORCID: 0009-0002-4545-3612\", \"ORCID: 0000-0002-5557-5798\", \"ORCID: 0009-0001-1725-2461\", \"ORCID: 0000-0003-1700-7834\"], \"AD\": [\"Department of Biochemistry and Biomedical Sciences, Michael G. DeGroote Institute for Infectious Disease Research, McMaster University, Hamilton, ON L8S 4L8, Canada.\", \"Department of Biochemistry and Biomedical Sciences, Michael G. DeGroote Institute for Infectious Disease Research, McMaster University, Hamilton, ON L8S 4L8, Canada.\", \"Department of Biochemistry and Biomedical Sciences, Michael G. DeGroote Institute for Infectious Disease Research, McMaster University, Hamilton, ON L8S 4L8, Canada.\", \"Department of Biochemistry and Biomedical Sciences, Michael G. DeGroote Institute for Infectious Disease Research, McMaster University, Hamilton, ON L8S 4L8, Canada.\", \"Department of Biochemistry and Biomedical Sciences, Michael G. DeGroote Institute for Infectious Disease Research, McMaster University, Hamilton, ON L8S 4L8, Canada.\", \"Department of Biochemistry and Biomedical Sciences, Michael G. DeGroote Institute for Infectious Disease Research, McMaster University, Hamilton, ON L8S 4L8, Canada.\"], \"LA\": [\"eng\"], \"GR\": [\"CRC-2017-00191/Canadian Government | Canadian Institutes of Health Research\", \"(CIHR)/\", \"INV-007414/GATES/Gates Foundation/United States\", \"GRFN RGPIN-2014-05577/Canadian Government | Natural Sciences and Engineering\", \"Research Council of Canada (NSERC)/\", \"INV-007414/GATES/Gates Foundation/United States\", \"INV-064444/Bill and Melinda Gates Foundation (GF)/\"], \"PT\": [\"Journal Article\"], \"DEP\": \"20250407\", \"PL\": \"United States\", \"TA\": \"Proc Natl Acad Sci U S A\", \"JT\": \"Proceedings of the National Academy of Sciences of the United States of America\", \"JID\": \"7505876\", \"RN\": [\"0 (Bacterial Proteins)\"], \"SB\": \"IM\", \"MH\": [\"*Bacteria/metabolism/enzymology/genetics\", \"Metabolic Networks and Pathways\", \"Deep Learning\", \"Bacterial Proteins/metabolism/genetics\", \"*Artificial Intelligence\", \"Computational Biology/methods\"], \"PMC\": \"PMC12012490\", \"OTO\": [\"NOTNLM\"], \"OT\": [\"Graphormer\", \"Transformer\", \"artificial intelligence\", \"bacterial metabolism\", \"knowledge graph\"], \"COIS\": [\"Competing interests statement:The authors declare no competing interest.\"], \"EDAT\": \"2025/04/07 18:24\", \"MHDA\": \"2025/04/07 18:25\", \"PMCR\": [\"2025/10/07\"], \"CRDT\": [\"2025/04/07 15:03\"], \"PHST\": [\"2025/04/07 18:25 [medline]\", \"2025/04/07 18:24 [pubmed]\", \"2025/04/07 15:03 [entrez]\", \"2025/10/07 00:00 [pmc-release]\"], \"AID\": [\"202425048 [pii]\", \"10.1073/pnas.2425048122 [doi]\"], \"PST\": \"ppublish\", \"SO\": \"Proc Natl Acad Sci U S A. 2025 Apr 15;122(15):e2425048122. doi: 10.1073/pnas.2425048122. Epub 2025 Apr 7.\"}, {\"PMID\": \"40161641\", \"OWN\": \"NLM\", \"STAT\": \"PubMed-not-MEDLINE\", \"LR\": \"20250916\", \"IS\": \"2692-8205 (Electronic) 2692-8205 (Linking)\", \"DP\": \"2025 May 29\", \"TI\": \"PloverDB: A high-performance platform for serving biomedical knowledge graphs as standards-compliant web APIs.\", \"LID\": \"2025.03.09.642156 [pii] 10.1101/2025.03.09.642156 [doi]\", \"AB\": \"Knowledge graphs are increasingly being used to integrate heterogeneous biomedical knowledge and data. General-purpose graph database management systems such as Neo4j are often used to host and search knowledge graphs, but such tools come with overhead and leave biomedical-specific standards compliance and reasoning to the user. Interoperability across biomedical knowledge bases and reasoning systems necessitates the use of standards such as those adopted by the Biomedical Data Translator consortium. We present PloverDB, a comprehensive software platform for hosting and efficiently serving biomedical knowledge graphs as standards-compliant web application programming interfaces. In addition to fundamental back-end knowledge reasoning tasks, PloverDB automatically handles entity resolution, exposure of standardized metadata and test data, and multiplexing of knowledge graphs, all in a single platform designed specifically for efficient query answering and ease of deployment. PloverDB increases data accessibility and utility by allowing data providers to quickly serve their biomedical knowledge graphs as standards-compliant web services. AVAILABILITY AND IMPLEMENTATION: PloverDB's source code and technical documentation are publicly available under an MIT License at github:RTXteam/PloverDB, archived on Zenodo at doi:10.5281/zenodo.15454600.\", \"FAU\": [\"Glen, Amy K\", \"Deutsch, Eric W\", \"Ramsey, Stephen A\"], \"AU\": [\"Glen AK\", \"Deutsch EW\", \"Ramsey SA\"], \"AUID\": [\"ORCID: 0000-0003-3691-2290\", \"ORCID: 0000-0001-8732-0928\", \"ORCID: 0000-0002-2168-5403\"], \"AD\": [\"School of Electrical Engineering and Computer Science, Oregon State University, Corvallis, Oregon, USA.\", \"Institute for Systems Biology, Seattle, Washington, USA.\", \"School of Electrical Engineering and Computer Science, Oregon State University, Corvallis, Oregon, USA.\", \"Department of Biomedical Sciences, Oregon State University, Corvallis, Oregon, USA.\"], \"LA\": [\"eng\"], \"GR\": [\"OT2 TR002520/TR/NCATS NIH HHS/United States\", \"OT2 TR003428/TR/NCATS NIH HHS/United States\"], \"PT\": [\"Journal Article\", \"Preprint\"], \"DEP\": \"20250529\", \"PL\": \"United States\", \"TA\": \"bioRxiv\", \"JT\": \"bioRxiv : the preprint server for biology\", \"JID\": \"101680187\", \"UIN\": [\"Bioinformatics. 2025 Jul 1;41(7):btaf380. doi: 10.1093/bioinformatics/btaf380.\", \"PMID: 40580448\"], \"PMC\": \"PMC11952354\", \"COIS\": [\"5Competing interests No competing interest is declared.\"], \"EDAT\": \"2025/03/31 21:17\", \"MHDA\": \"2025/03/31 21:18\", \"PMCR\": [\"2025/06/02\"], \"CRDT\": [\"2025/03/31 06:22\"], \"PHST\": [\"2025/03/31 21:17 [pubmed]\", \"2025/03/31 21:18 [medline]\", \"2025/03/31 06:22 [entrez]\", \"2025/06/02 00:00 [pmc-release]\"], \"AID\": [\"2025.03.09.642156 [pii]\", \"10.1101/2025.03.09.642156 [doi]\"], \"PST\": \"epublish\", \"SO\": \"bioRxiv [Preprint]. 2025 May 29:2025.03.09.642156. doi: 10.1101/2025.03.09.642156.\"}, {\"PMID\": \"40062257\", \"OWN\": \"NLM\", \"STAT\": \"PubMed-not-MEDLINE\", \"LR\": \"20250310\", \"IS\": \"2376-5992 (Electronic) 2376-5992 (Linking)\", \"VI\": \"11\", \"DP\": \"2025\", \"TI\": \"Construction of a user-friendly software-defined networking management using a graph-based abstraction layer.\", \"PG\": \"e2674\", \"LID\": \"10.7717/peerj-cs.2674 [doi] e2674\", \"AB\": \"The software-defined networking (SDN) paradigm relies on the decoupling of the control plane and data plane. Northbound interfaces enable the implementation of network services through logical centralised control. Suitable northbound interfaces and application-oriented abstractions are the core of the SDN ecosystem. This article presents an architecture to represent the network as a graph. The purpose of this architecture is to implement an abstraction of the SDN controller at the application plane. We abstract all network elements using a graph model, with the attributes of the elements as the attributes of the graph. This virtualized logical abstraction layer, which is not limited by the physical network, enables network administrators to schedule network resources directly in a global view. The feasibility of the presented graph abstraction was verified through experiments in topological display, dynamic route, access control, and data persistence. The performance of the shortest path in the graph-based abstraction layer and graph database proves the necessity of the graph abstraction layer. Empirical evidence demonstrates that the graph-based abstraction layer can facilitate network slicing, maintain a dependable depiction of the real network, streamline network administration and network application development, and provide a sophisticated abstraction that is easily understandable to network administrators.\", \"CI\": [\"(c) 2025 Jia et al.\"], \"FAU\": [\"Jia, Yufeng\", \"Ren, Jiadong\", \"Li, Xianshan\", \"He, Haitao\", \"Zhang, Pengwei\", \"Li, Rong\"], \"AU\": [\"Jia Y\", \"Ren J\", \"Li X\", \"He H\", \"Zhang P\", \"Li R\"], \"AD\": [\"School of Information Science and Engineering, Xinjiang College of Science and Technology, Korla, Xinjiang, China.\", \"School of Information Science and Engineering, Yanshan University, Qinhuangdao, Hebei, China.\", \"School of Information Science and Engineering, Yanshan University, Qinhuangdao, Hebei, China.\", \"School of Information Science and Engineering, Yanshan University, Qinhuangdao, Hebei, China.\", \"School of Information Science and Engineering, Yanshan University, Qinhuangdao, Hebei, China.\", \"School of Information Science and Engineering, Xinjiang College of Science and Technology, Korla, Xinjiang, China.\", \"School of Information Science and Engineering, Yanshan University, Qinhuangdao, Hebei, China.\", \"School of Information Science and Engineering, Xinjiang College of Science and Technology, Korla, Xinjiang, China.\"], \"LA\": [\"eng\"], \"PT\": [\"Journal Article\"], \"DEP\": \"20250214\", \"PL\": \"United States\", \"TA\": \"PeerJ Comput Sci\", \"JT\": \"PeerJ. Computer science\", \"JID\": \"101660598\", \"PMC\": \"PMC11888870\", \"OTO\": [\"NOTNLM\"], \"OT\": [\"Abstraction layer\", \"Graph model\", \"Network application management\", \"Software-defined networking (SDN)\"], \"COIS\": [\"The authors declare that they have no competing interests.\"], \"EDAT\": \"2025/03/10 17:18\", \"MHDA\": \"2025/03/10 17:19\", \"PMCR\": [\"2025/02/14\"], \"CRDT\": [\"2025/03/10 06:41\"], \"PHST\": [\"2024/09/09 00:00 [received]\", \"2025/01/08 00:00 [accepted]\", \"2025/03/10 17:19 [medline]\", \"2025/03/10 17:18 [pubmed]\", \"2025/03/10 06:41 [entrez]\", \"2025/02/14 00:00 [pmc-release]\"], \"AID\": [\"cs-2674 [pii]\", \"10.7717/peerj-cs.2674 [doi]\"], \"PST\": \"epublish\", \"SO\": \"PeerJ Comput Sci. 2025 Feb 14;11:e2674. doi: 10.7717/peerj-cs.2674. eCollection 2025.\"}, {\"PMID\": \"40057225\", \"OWN\": \"NLM\", \"STAT\": \"MEDLINE\", \"DCOM\": \"20250512\", \"LR\": \"20250512\", \"IS\": \"1089-8638 (Electronic) 0022-2836 (Linking)\", \"VI\": \"437\", \"IP\": \"10\", \"DP\": \"2025 May 15\", \"TI\": \"Yeast Knowledge Graphs Database for Exploring Saccharomyces Cerevisiae and Schizosaccharomyces Pombe.\", \"PG\": \"169072\", \"LID\": \"S0022-2836(25)00138-X [pii] 10.1016/j.jmb.2025.169072 [doi]\", \"AB\": \"Biomedical literature contains an extensive wealth of information on gene and protein function across various biological processes and diseases. However, navigating this vast and often restricted-access data can be challenging, making it difficult to extract specific insights efficiently. In this study, we introduce a high-throughput pipeline that leverages OpenAI's Generative Pre-Trained Transformer Model (GPT) to automate the extraction and analysis of gene function information. We applied this approach to 84,427 publications on Saccharomyces cerevisiae and 6,452 publications on Schizosaccharomyces pombe, identifying 3,432,749 relationships for budding yeast and 421,198 relationships for S. pombe. This resulted in a comprehensive, searchable online Knowledge Graph database, available at yeast.connectome.tools and spombe.connectome.tools, which offers users extensive access to various interactions and pathways. Our analysis underscores the power of integrating artificial intelligence with bioinformatics, as demonstrated through key insights into important nodes like Hsp104 and Atg8 proteins. This work not only facilitates efficient data extraction in yeast research but also presents a scalable model for similar studies in other biological systems.\", \"CI\": [\"Copyright (c) 2025 The Author(s). Published by Elsevier Ltd.. All rights reserved.\"], \"FAU\": [\"Kumar, Mani R\", \"Arulprakasam, Karthick Raja\", \"Kutevska, An-Nikol\", \"Mutwil, Marek\", \"Thibault, Guillaume\"], \"AU\": [\"Kumar MR\", \"Arulprakasam KR\", \"Kutevska AN\", \"Mutwil M\", \"Thibault G\"], \"AD\": [\"School of Biological Sciences, Nanyang Technological University, 60 Nanyang Drive, Singapore 637551 Singapore.\", \"School of Biological Sciences, Nanyang Technological University, 60 Nanyang Drive, Singapore 637551 Singapore.\", \"School of Biological Sciences, Nanyang Technological University, 60 Nanyang Drive, Singapore 637551 Singapore.\", \"School of Biological Sciences, Nanyang Technological University, 60 Nanyang Drive, Singapore 637551 Singapore; Department of Plant and Environmental Sciences, University of Copenhagen, Thorvaldsensvej 40, 1871 Frederiksberg C, Copenhagen, Denmark. Electronic address: mutwil@plen.ku.dk.\", \"School of Biological Sciences, Nanyang Technological University, 60 Nanyang Drive, Singapore 637551 Singapore; Mechanobiology Institute, National University of Singapore, Singapore 117411 Singapore. Electronic address: thibault@ntu.edu.sg.\"], \"LA\": [\"eng\"], \"PT\": [\"Journal Article\"], \"DEP\": \"20250306\", \"PL\": \"Netherlands\", \"TA\": \"J Mol Biol\", \"JT\": \"Journal of molecular biology\", \"JID\": \"2985088R\", \"SB\": \"IM\", \"MH\": [\"*Schizosaccharomyces/genetics/metabolism\", \"*Saccharomyces cerevisiae/genetics/metabolism\", \"*Computational Biology/methods\", \"*Databases, Genetic\"], \"OTO\": [\"NOTNLM\"], \"OT\": [\"Saccharomyces\", \"Schizosaccharomyces\", \"bioinformatics\", \"knowledge graph\", \"yeast\"], \"COIS\": [\"Declaration of competing interest The authors declare the following financial\", \"interests/personal relationships which may be considered as potential competing\", \"interests: \\\"Guillaume Thibault reports financial support was provided by Nanyang\", \"Technological University. If there are other authors, they declare that they have\", \"no known competing financial interests or personal relationships that could have\", \"appeared to influence the work reported in this paper\\\".\"], \"EDAT\": \"2025/03/09 15:10\", \"MHDA\": \"2025/04/03 18:25\", \"CRDT\": [\"2025/03/08 19:18\"], \"PHST\": [\"2024/12/05 00:00 [received]\", \"2025/02/28 00:00 [revised]\", \"2025/03/03 00:00 [accepted]\", \"2025/04/03 18:25 [medline]\", \"2025/03/09 15:10 [pubmed]\", \"2025/03/08 19:18 [entrez]\"], \"AID\": [\"S0022-2836(25)00138-X [pii]\", \"10.1016/j.jmb.2025.169072 [doi]\"], \"PST\": \"ppublish\", \"SO\": \"J Mol Biol. 2025 May 15;437(10):169072. doi: 10.1016/j.jmb.2025.169072. Epub 2025 Mar 6.\"}, {\"PMID\": \"40001634\", \"OWN\": \"NLM\", \"STAT\": \"PubMed-not-MEDLINE\", \"LR\": \"20250228\", \"IS\": \"2306-5354 (Print) 2306-5354 (Electronic) 2306-5354 (Linking)\", \"VI\": \"12\", \"IP\": \"2\", \"DP\": \"2025 Jan 26\", \"TI\": \"Storage and Query of Drug Knowledge Graphs Using Distributed Graph Databases: A Case Study.\", \"LID\": \"10.3390/bioengineering12020115 [doi] 115\", \"AB\": \"BACKGROUND: Distributed graph databases are a promising method for storing and conducting complex pathway queries on large-scale drug knowledge graphs to support drug research. However, there is a research gap in evaluating drug knowledge graphs' storage and query performance based on distributed graph databases. This study evaluates the feasibility and performance of distributed graph databases in managing large-scale drug knowledge graphs. METHODS: First, a drug knowledge graph storage and query system is designed based on the Nebula Graph database. Second, the system's writing and query performance is evaluated. Finally, two drug repurposing benchmarks are used to provide a more extensive and reliable assessment. RESULTS: The performance of distributed graph databases surpasses that of single-machine databases, including data writing, regular queries, constrained queries, and concurrent queries. Additionally, the advantages of distributed graph databases in writing performance become more pronounced as the data volume increases. The query performance benefits of distributed graph databases also improve with the complexity of query tasks. The drug repurposing evaluation results show that 78.54% of the pathways are consistent with currently approved drug treatments according to repoDB. Additionally, 12 potential pathways for new drug indications are found to have literature support according to DrugRepoBank. CONCLUSIONS: The proposed system is able to construct, store, and query a large graph of multisource drug knowledge and provides reliable and explainable drug-disease paths for drug repurposing.\", \"FAU\": [\"Han, Xingjian\", \"Tian, Yu\"], \"AU\": [\"Han X\", \"Tian Y\"], \"AD\": [\"China Electronic Product Reliability and Environmental Testing Research Institute (The Fifth Electronic Research Institute of MIIT), Guangzhou 510610, China.\", \"Engineering Research Center of EMR and Intelligent Expert Systems, Ministry of Education, Key Laboratory for Biomedical Engineering of Ministry of Education, College of Biomedical Engineering and Instrument Science, Zhejiang University, Hangzhou 310027, China.\"], \"LA\": [\"eng\"], \"GR\": [\"82172069/National Natural Science Foundation of China/\"], \"PT\": [\"Journal Article\"], \"DEP\": \"20250126\", \"PL\": \"Switzerland\", \"TA\": \"Bioengineering (Basel)\", \"JT\": \"Bioengineering (Basel, Switzerland)\", \"JID\": \"101676056\", \"PMC\": \"PMC11852034\", \"OTO\": [\"NOTNLM\"], \"OT\": [\"distributed graph databases\", \"drug knowledge graph\", \"real-time data retrieval\", \"storage and query\"], \"COIS\": [\"The authors declare no conflicts of interest.\"], \"EDAT\": \"2025/02/26 06:28\", \"MHDA\": \"2025/02/26 06:29\", \"PMCR\": [\"2025/01/26\"], \"CRDT\": [\"2025/02/26 01:03\"], \"PHST\": [\"2024/12/13 00:00 [received]\", \"2025/01/21 00:00 [revised]\", \"2025/01/25 00:00 [accepted]\", \"2025/02/26 06:29 [medline]\", \"2025/02/26 06:28 [pubmed]\", \"2025/02/26 01:03 [entrez]\", \"2025/01/26 00:00 [pmc-release]\"], \"AID\": [\"bioengineering12020115 [pii]\", \"bioengineering-12-00115 [pii]\", \"10.3390/bioengineering12020115 [doi]\"], \"PST\": \"epublish\", \"SO\": \"Bioengineering (Basel). 2025 Jan 26;12(2):115. doi: 10.3390/bioengineering12020115.\"}, {\"PMID\": \"39961540\", \"OWN\": \"NLM\", \"STAT\": \"MEDLINE\", \"DCOM\": \"20250508\", \"LR\": \"20250508\", \"IS\": \"1532-0480 (Electronic) 1532-0464 (Print) 1532-0464 (Linking)\", \"VI\": \"163\", \"DP\": \"2025 Mar\", \"TI\": \"Developing libraries of semantically-augmented graphics as visual standards for biomedical information systems.\", \"PG\": \"104804\", \"LID\": \"S1532-0464(25)00033-4 [pii] 10.1016/j.jbi.2025.104804 [doi]\", \"AB\": \"OBJECTIVE: Visual representations generally serve as supplements to information, rather than as bearers of computable information themselves. Our objective is to develop a method for creating semantically-augmented graphic libraries that will serve as visual standards and can be implemented as visual assets in intelligent information systems. METHODS: Graphics were developed using a composable approach and specified using SVG. OWL was used to represent the entities of our system, which include elements, units, graphics, graphic libraries, and library collections. A graph database serves as our data management system. Semantics are applied at multiple levels: (a) each element is associated with a semantic style class to link visual style to semantic meaning, (b) graphics are described using object properties and data properties, (c) relationships are specified between graphics, and (d) mappings are made between the graphics and outside resources. RESULTS: The Graphic Library web application enables users to browse the libraries, view information pages for each graphic, and download individual graphics. We demonstrate how SPARQL can be employed to query the graphics database and the APIs can be used to retrieve the graphics and associated data for applications. In addition, this work shows that our method of designing composable graphics is well-suited to depicting variations in human anatomy. CONCLUSION: This work provides a bridge between visual communication and the field of knowledge representation. We demonstrate a method for creating visual standards that are compatible with practices in biomedical ontology and implement a system for making them accessible to information systems.\", \"CI\": [\"Copyright (c) 2025. Published by Elsevier Inc.\"], \"FAU\": [\"Clarkson, Melissa D\", \"Roggenkamp, Steven\", \"Detwiler, Landon T\"], \"AU\": [\"Clarkson MD\", \"Roggenkamp S\", \"Detwiler LT\"], \"AD\": [\"Division of Biomedical Informatics, University of Kentucky, Multidisciplinary Science Building, Room 230C, Lexington, KY 40506, USA. Electronic address: mclarkson@uky.edu.\", \"Division of Biomedical Informatics, University of Kentucky, Multidisciplinary Science Building, Room 230C, Lexington, KY 40506, USA. Electronic address: steve.roggenkamp@uky.edu.\", \"Division of Biomedical Informatics, University of Kentucky, Multidisciplinary Science Building, Room 230C, Lexington, KY 40506, USA. Electronic address: todd.detwiler@uky.edu.\"], \"LA\": [\"eng\"], \"GR\": [\"R01 DE030100/DE/NIDCR NIH HHS/United States\"], \"PT\": [\"Journal Article\"], \"DEP\": \"20250216\", \"PL\": \"United States\", \"TA\": \"J Biomed Inform\", \"JT\": \"Journal of biomedical informatics\", \"JID\": \"100970413\", \"SB\": \"IM\", \"MH\": [\"*Semantics\", \"*Computer Graphics/standards\", \"*Medical Informatics/methods\", \"Humans\", \"User-Computer Interface\", \"Databases, Factual\"], \"PMC\": \"PMC11899390\", \"MID\": [\"NIHMS2060003\"], \"OTO\": [\"NOTNLM\"], \"OT\": [\"Ontology\", \"Semantic web\", \"Standards\", \"Visual communication\"], \"COIS\": [\"Declaration of competing interest The authors declare that they have no known\", \"competing financial interests or personal relationships that could have appeared\", \"to influence the work reported in this paper.\"], \"EDAT\": \"2025/02/18 00:22\", \"MHDA\": \"2025/03/03 00:54\", \"PMCR\": [\"2026/03/01\"], \"CRDT\": [\"2025/02/17 19:28\"], \"PHST\": [\"2024/10/31 00:00 [received]\", \"2025/02/09 00:00 [revised]\", \"2025/02/14 00:00 [accepted]\", \"2026/03/01 00:00 [pmc-release]\", \"2025/03/03 00:54 [medline]\", \"2025/02/18 00:22 [pubmed]\", \"2025/02/17 19:28 [entrez]\"], \"AID\": [\"S1532-0464(25)00033-4 [pii]\", \"10.1016/j.jbi.2025.104804 [doi]\"], \"PST\": \"ppublish\", \"SO\": \"J Biomed Inform. 2025 Mar;163:104804. doi: 10.1016/j.jbi.2025.104804. Epub 2025 Feb 16.\"}, {\"PMID\": \"39906190\", \"OWN\": \"NLM\", \"STAT\": \"PubMed-not-MEDLINE\", \"LR\": \"20250206\", \"IS\": \"2624-909X (Electronic) 2624-909X (Linking)\", \"VI\": \"7\", \"DP\": \"2024\", \"TI\": \"A scalable tool for analyzing genomic variants of humans using knowledge graphs and graph machine learning.\", \"PG\": \"1466391\", \"LID\": \"10.3389/fdata.2024.1466391 [doi] 1466391\", \"AB\": \"Advances in high-throughput genome sequencing have enabled large-scale genome sequencing in clinical practice and research studies. By analyzing genomic variants of humans, scientists can gain better understanding of the risk factors of complex diseases such as cancer and COVID-19. To model and analyze the rich genomic data, knowledge graphs (KGs) and graph machine learning (GML) can be regarded as enabling technologies. In this article, we present a scalable tool called VariantKG for analyzing genomic variants of humans modeled using KGs and GML. Specifically, we used publicly available genome sequencing data from patients with COVID-19. VariantKG extracts variant-level genetic information output by a variant calling pipeline, annotates the variant data with additional metadata, and converts the annotated variant information into a KG represented using the Resource Description Framework (RDF). The resulting KG is further enhanced with patient metadata and stored in a scalable graph database that enables efficient RDF indexing and query processing. VariantKG employs the Deep Graph Library (DGL) to perform GML tasks such as node classification. A user can extract a subset of the KG and perform inference tasks using DGL. The user can monitor the training and testing performance and hardware utilization. We tested VariantKG for KG construction by using 1,508 genome sequences, leading to 4 billion RDF statements. We evaluated GML tasks using VariantKG by selecting a subset of 500 sequences from the KG and performing node classification using well-known GML techniques such as GraphSAGE, Graph Convolutional Network (GCN) and Graph Transformer. VariantKG has intuitive user interfaces and features enabling a low barrier to entry for KG construction, model inference, and model interpretation on genomic variants of humans.\", \"CI\": [\"Copyright (c) 2025 Prasanna, Kumar, Rao, Simoes and Rao.\"], \"FAU\": [\"Prasanna, Shivika\", \"Kumar, Ajay\", \"Rao, Deepthi\", \"Simoes, Eduardo J\", \"Rao, Praveen\"], \"AU\": [\"Prasanna S\", \"Kumar A\", \"Rao D\", \"Simoes EJ\", \"Rao P\"], \"AD\": [\"Department of Electrical Engineering and Computer Science, University of Missouri, Columbia, MO, United States.\", \"Department of Electrical Engineering and Computer Science, University of Missouri, Columbia, MO, United States.\", \"Department of Pathology and Anatomical Sciences, University of Missouri, Columbia, MO, United States.\", \"Department of Biomedical Informatics, Biostatistics and Medical Epidemiology, University of Missouri, Columbia, MO, United States.\", \"Department of Electrical Engineering and Computer Science, University of Missouri, Columbia, MO, United States.\"], \"LA\": [\"eng\"], \"PT\": [\"Journal Article\"], \"DEP\": \"20250121\", \"PL\": \"Switzerland\", \"TA\": \"Front Big Data\", \"JT\": \"Frontiers in big data\", \"JID\": \"101770603\", \"PMC\": \"PMC11790625\", \"OTO\": [\"NOTNLM\"], \"OT\": [\"graph machine learning\", \"human genomic variants\", \"inference\", \"knowledge graphs\", \"scalability\"], \"COIS\": [\"The authors declare that the research was conducted in the absence of any\", \"commercial or financial relationships that could be construed as a potential\", \"conflict of interest. The author(s) declared that they were an editorial board\", \"member of Frontiers, at the time of submission. This had no impact on the peer\", \"review process and the final decision.\"], \"EDAT\": \"2025/02/05 06:20\", \"MHDA\": \"2025/02/05 06:21\", \"PMCR\": [\"2025/01/21\"], \"CRDT\": [\"2025/02/05 04:16\"], \"PHST\": [\"2024/07/17 00:00 [received]\", \"2024/12/12 00:00 [accepted]\", \"2025/02/05 06:21 [medline]\", \"2025/02/05 06:20 [pubmed]\", \"2025/02/05 04:16 [entrez]\", \"2025/01/21 00:00 [pmc-release]\"], \"AID\": [\"10.3389/fdata.2024.1466391 [doi]\"], \"PST\": \"epublish\", \"SO\": \"Front Big Data. 2025 Jan 21;7:1466391. doi: 10.3389/fdata.2024.1466391. eCollection 2024.\"}, {\"PMID\": \"39905013\", \"OWN\": \"NLM\", \"STAT\": \"MEDLINE\", \"DCOM\": \"20250205\", \"LR\": \"20250609\", \"IS\": \"2052-4463 (Electronic) 2052-4463 (Linking)\", \"VI\": \"12\", \"IP\": \"1\", \"DP\": \"2025 Feb 4\", \"TI\": \"A Chinese Knowledge Graph Dataset in the Field of Scientific Fitness.\", \"PG\": \"205\", \"LID\": \"10.1038/s41597-025-04519-6 [doi] 205\", \"AB\": \"To promote the development of scientific fitness research and practice, we propose the Chinese Knowledge Graph Dataset in the Field of Scientific Fitness (FitKG-CN). This knowledge graph contains over 10,000 fitness-related terms, categorized into eight main groups: body parts, items of exercise, fitness movement, equipment and tools, exercise goals, anatomical structures, nutrients, and technical terms. The construction of FitKG-CN is based on authoritative data sources, undergoing rigorous preprocessing, including noise removal, format standardization, and normalization of entities and relationships. The data is manually annotated on a professional platform and ultimately stored in a Neo4j graph database for visualization. Additionally, we trained a Chinese SpERT model using the manually annotated data to enhance the automation of data processing. The experimental results show that the model achieved an F1 score of 94.05% in entity recognition tasks and 82.00% in relation extraction tasks, validating the effectiveness of the model and improving the scalability of the dataset.\", \"CI\": [\"(c) 2025. The Author(s).\"], \"FAU\": [\"Du, Shutong\", \"Liu, Zhitong\", \"Pan, Bingyu\"], \"AU\": [\"Du S\", \"Liu Z\", \"Pan B\"], \"AUID\": [\"ORCID: 0009-0004-4139-3776\"], \"AD\": [\"School of Sports Engineering, Beijing Sport University, Beijing, China.\", \"China Sports Big Data Center, Beijing Sport University, Beijing, China.\", \"School of Sports Engineering, Beijing Sport University, Beijing, China.\", \"China Sports Big Data Center, Beijing Sport University, Beijing, China.\", \"School of Sports Engineering, Beijing Sport University, Beijing, China. panbingyu@bsu.edu.cn.\", \"China Sports Big Data Center, Beijing Sport University, Beijing, China. panbingyu@bsu.edu.cn.\"], \"LA\": [\"eng\"], \"PT\": [\"Dataset\", \"Journal Article\"], \"DEP\": \"20250204\", \"PL\": \"England\", \"TA\": \"Sci Data\", \"JT\": \"Scientific data\", \"JID\": \"101640192\", \"SB\": \"IM\", \"MH\": [\"Humans\", \"China\", \"*Databases, Factual\", \"Exercise\"], \"PMC\": \"PMC11794866\", \"COIS\": [\"Competing interests: The authors declare no competing interests.\"], \"EDAT\": \"2025/02/05 06:21\", \"MHDA\": \"2025/02/05 06:22\", \"PMCR\": [\"2025/02/04\"], \"CRDT\": [\"2025/02/05 00:05\"], \"PHST\": [\"2024/10/07 00:00 [received]\", \"2025/01/23 00:00 [accepted]\", \"2025/02/05 06:22 [medline]\", \"2025/02/05 06:21 [pubmed]\", \"2025/02/05 00:05 [entrez]\", \"2025/02/04 00:00 [pmc-release]\"], \"AID\": [\"10.1038/s41597-025-04519-6 [pii]\", \"4519 [pii]\", \"10.1038/s41597-025-04519-6 [doi]\"], \"PST\": \"epublish\", \"SO\": \"Sci Data. 2025 Feb 4;12(1):205. doi: 10.1038/s41597-025-04519-6.\"}]\n"
          ]
        }
      ],
      "source": [
        "print(json.dumps(record_list))"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 10,
      "id": "c8155faa-104e-4998-8bd4-342ec6486897",
      "metadata": {
        "id": "c8155faa-104e-4998-8bd4-342ec6486897"
      },
      "outputs": [],
      "source": [
        "# Writing the dictionary to a JSON file\n",
        "with open(r'content/pubmed_search_graph_theory.json', 'w', encoding='utf-8') as fp:\n",
        "    fp.write(json.dumps(record_list))"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 11,
      "id": "c866e13e-da2c-4340-b65d-8af02c6caed0",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "c866e13e-da2c-4340-b65d-8af02c6caed0",
        "outputId": "668a9a0c-458c-4dfd-dc21-982154d99bef"
      },
      "outputs": [
        {
          "output_type": "execute_result",
          "data": {
            "text/plain": [
              "[{'PMID': '41425057',\n",
              "  'OT': ['Neo4j',\n",
              "   'Text2Cypher',\n",
              "   'cypher query',\n",
              "   'graph database',\n",
              "   'knowledge graphs',\n",
              "   'large language models',\n",
              "   'nature finance',\n",
              "   'question answering']},\n",
              " {'PMID': '41369777', 'OT': []},\n",
              " {'PMID': '41292650',\n",
              "  'OT': ['BERT embeddings',\n",
              "   'Case reports',\n",
              "   'PubMed',\n",
              "   'data mining',\n",
              "   'evidence based medicine',\n",
              "   'graph database',\n",
              "   'network analysis',\n",
              "   'semantic similarity']},\n",
              " {'PMID': '41252800',\n",
              "  'OT': ['Cancer gene networks',\n",
              "   'Gene pathway fingerprint',\n",
              "   'Identifying cancer drug targets',\n",
              "   'Matrix of genes and reactions',\n",
              "   'Pathways shared in cancer and comorbid Mendelian Diseases',\n",
              "   'Targeted cancer therapy combinations']},\n",
              " {'PMID': '41173963',\n",
              "  'OT': ['Gear shaving tooth profile cutting depth error',\n",
              "   'Knowledge graph',\n",
              "   'Shaving-induced mid-profile concavity error',\n",
              "   'Siloization issue',\n",
              "   'Tooth profile error']},\n",
              " {'PMID': '41139924', 'OT': ['EGNF', 'biomarker', 'graph neural network']},\n",
              " {'PMID': '41030660',\n",
              "  'OT': ['Broiler farming',\n",
              "   'Extraction of knowledge',\n",
              "   'Knowledge graph',\n",
              "   'Large language model']},\n",
              " {'PMID': '40968918',\n",
              "  'OT': ['graph databases',\n",
              "   'knowledge retrieval',\n",
              "   'large language models',\n",
              "   'orbital debris',\n",
              "   'retrieval-augmented generation',\n",
              "   'space situational awareness']},\n",
              " {'PMID': '40804424',\n",
              "  'OT': ['Bioactive compounds',\n",
              "   'Clinical trials',\n",
              "   'Data integration',\n",
              "   'Evidence-based health',\n",
              "   'Knowledge graph',\n",
              "   'Natural language processing (NLP)',\n",
              "   'PubMed']},\n",
              " {'PMID': '40700523', 'OT': []},\n",
              " {'PMID': '40648170',\n",
              "  'OT': ['BiLSTM (bidirectional long short-term memory network)',\n",
              "   'entity recognition',\n",
              "   'fault diagnosis',\n",
              "   'intelligent production line',\n",
              "   'knowledge graph']},\n",
              " {'PMID': '40640288',\n",
              "  'OT': ['Gas polyethylene pipes',\n",
              "   'Knowledge extraction',\n",
              "   'Knowledge graph',\n",
              "   'Named entity recognition']},\n",
              " {'PMID': '40580451', 'OT': []},\n",
              " {'PMID': '40580448', 'OT': []},\n",
              " {'PMID': '40518789',\n",
              "  'OT': ['ZHOU Meisheng',\n",
              "   'data mining',\n",
              "   'knowledge graph',\n",
              "   'low back pain',\n",
              "   'moxibustion method']},\n",
              " {'PMID': '40357433',\n",
              "  'OT': ['EKG-CMG',\n",
              "   'cardiovascular medication knowledge',\n",
              "   'expert knowledge graph',\n",
              "   'knowledge discover and reasoning',\n",
              "   'multidimensional relationships',\n",
              "   'ontology construction']},\n",
              " {'PMID': '40292999',\n",
              "  'OT': ['Internet of Things (IoT)',\n",
              "   'cloud computing',\n",
              "   'graph encryption',\n",
              "   'graph intersection',\n",
              "   'privacy preserving',\n",
              "   'wireless communications']},\n",
              " {'PMID': '40193601',\n",
              "  'OT': ['Graphormer',\n",
              "   'Transformer',\n",
              "   'artificial intelligence',\n",
              "   'bacterial metabolism',\n",
              "   'knowledge graph']},\n",
              " {'PMID': '40161641', 'OT': []},\n",
              " {'PMID': '40062257',\n",
              "  'OT': ['Abstraction layer',\n",
              "   'Graph model',\n",
              "   'Network application management',\n",
              "   'Software-defined networking (SDN)']},\n",
              " {'PMID': '40057225',\n",
              "  'OT': ['Saccharomyces',\n",
              "   'Schizosaccharomyces',\n",
              "   'bioinformatics',\n",
              "   'knowledge graph',\n",
              "   'yeast']},\n",
              " {'PMID': '40001634',\n",
              "  'OT': ['distributed graph databases',\n",
              "   'drug knowledge graph',\n",
              "   'real-time data retrieval',\n",
              "   'storage and query']},\n",
              " {'PMID': '39961540',\n",
              "  'OT': ['Ontology', 'Semantic web', 'Standards', 'Visual communication']},\n",
              " {'PMID': '39906190',\n",
              "  'OT': ['graph machine learning',\n",
              "   'human genomic variants',\n",
              "   'inference',\n",
              "   'knowledge graphs',\n",
              "   'scalability']},\n",
              " {'PMID': '39905013', 'OT': []}]"
            ]
          },
          "metadata": {},
          "execution_count": 11
        }
      ],
      "source": [
        "# Extract only PMID and OT fields, handling cases where 'OT' might not exist\n",
        "extracted_data = [{\"PMID\": entry[\"PMID\"], \"OT\": entry.get(\"OT\", [])} for entry in record_list]\n",
        "\n",
        "# Display the extracted data\n",
        "extracted_data"
      ]
    },
    {
      "cell_type": "code",
      "execution_count": 12,
      "id": "ac56d97f-c54e-4b4c-9463-33b8f7d6607e",
      "metadata": {
        "id": "ac56d97f-c54e-4b4c-9463-33b8f7d6607e"
      },
      "outputs": [],
      "source": [
        "# Writing the dictionary to a JSON file\n",
        "with open(r'content/pubmed_search_graph_theory_kws.json', 'w', encoding='utf-8') as fp:\n",
        "    fp.write(json.dumps(extracted_data))"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "import json\n",
        "\n",
        "# File paths\n",
        "input_file = \"/content/pubmed_search_graph_theory_kws.json\"\n",
        "output_file = \"/content/pubmed_inserts.sql\"\n",
        "\n",
        "# Load JSON\n",
        "with open(input_file, \"r\", encoding=\"utf-8\") as f:\n",
        "    data = json.load(f)\n",
        "\n",
        "with open(output_file, \"w\", encoding=\"utf-8\") as f:\n",
        "    f.write(\"INSERT INTO pubmed_raw (PMID, OT) VALUES\\n\")\n",
        "\n",
        "    values = []\n",
        "\n",
        "    for record in data:\n",
        "        pmid = str(record.get(\"PMID\", \"\")).strip()\n",
        "        ot = record.get(\"OT\", [])\n",
        "\n",
        "        if pmid:\n",
        "            if not isinstance(ot, list):\n",
        "                ot = [ot]\n",
        "\n",
        "            # Convert list to JSON string\n",
        "            ot_json = json.dumps(ot, ensure_ascii=False)\n",
        "\n",
        "            # Escape single quotes for SQL safety\n",
        "            ot_json = ot_json.replace(\"'\", \"''\")\n",
        "\n",
        "            values.append(f\"('{pmid}', CAST('{ot_json}' AS JSON))\")\n",
        "\n",
        "    f.write(\",\\n\".join(values))\n",
        "    f.write(\";\\n\")\n",
        "\n",
        "print(\"Finished. SQL file saved to:\", output_file)"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "6z_5nlzzUGYL",
        "outputId": "64723202-589b-4868-c328-c407f273d073"
      },
      "id": "6z_5nlzzUGYL",
      "execution_count": 14,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "Finished. SQL file saved to: /content/pubmed_inserts.sql\n"
          ]
        }
      ]
    }
  ],
  "metadata": {
    "kernelspec": {
      "display_name": "Python 3 (ipykernel)",
      "language": "python",
      "name": "python3"
    },
    "language_info": {
      "codemirror_mode": {
        "name": "ipython",
        "version": 3
      },
      "file_extension": ".py",
      "mimetype": "text/x-python",
      "name": "python",
      "nbconvert_exporter": "python",
      "pygments_lexer": "ipython3",
      "version": "3.11.5"
    },
    "colab": {
      "provenance": [],
      "include_colab_link": true
    }
  },
  "nbformat": 4,
  "nbformat_minor": 5
}